{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "e4b6ddc4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Collecting pypi-cdecimal\n",
      "  Downloading pypi-cdecimal-2.3.tar.gz (90 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m90.6/90.6 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\n",
      "  \n",
      "  \u001b[31m×\u001b[0m \u001b[32mpython setup.py egg_info\u001b[0m did not run successfully.\n",
      "  \u001b[31m│\u001b[0m exit code: \u001b[1;36m1\u001b[0m\n",
      "  \u001b[31m╰─>\u001b[0m \u001b[31m[10 lines of output]\u001b[0m\n",
      "  \u001b[31m   \u001b[0m Traceback (most recent call last):\n",
      "  \u001b[31m   \u001b[0m   File \"<string>\", line 2, in <module>\n",
      "  \u001b[31m   \u001b[0m   File \"<pip-setuptools-caller>\", line 34, in <module>\n",
      "  \u001b[31m   \u001b[0m   File \"/tmp/pip-install-8a_oazqv/pypi-cdecimal_01f2bcda3d444cc188abd520e7b8ef3b/setup.py\", line 426, in <module>\n",
      "  \u001b[31m   \u001b[0m     ext_modules = [cdecimal_ext(MACHINE)]\n",
      "  \u001b[31m   \u001b[0m   File \"/tmp/pip-install-8a_oazqv/pypi-cdecimal_01f2bcda3d444cc188abd520e7b8ef3b/setup.py\", line 315, in cdecimal_ext\n",
      "  \u001b[31m   \u001b[0m     config_vars = configure(machine, cc, py_size_t)\n",
      "  \u001b[31m   \u001b[0m   File \"/tmp/pip-install-8a_oazqv/pypi-cdecimal_01f2bcda3d444cc188abd520e7b8ef3b/setup.py\", line 230, in configure\n",
      "  \u001b[31m   \u001b[0m     os.chmod(\"./configure\", 0x1ed) # pip removes execute permissions.\n",
      "  \u001b[31m   \u001b[0m FileNotFoundError: [Errno 2] No such file or directory: './configure'\n",
      "  \u001b[31m   \u001b[0m \u001b[31m[end of output]\u001b[0m\n",
      "  \n",
      "  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "\u001b[1;31merror\u001b[0m: \u001b[1mmetadata-generation-failed\u001b[0m\n",
      "\n",
      "\u001b[31m×\u001b[0m Encountered error while generating package metadata.\n",
      "\u001b[31m╰─>\u001b[0m See above for output.\n",
      "\n",
      "\u001b[1;35mnote\u001b[0m: This is an issue with the package mentioned above, not pip.\n",
      "\u001b[1;36mhint\u001b[0m: See above for details.\n",
      "\u001b[?25hDefaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: biopython in /home/maxim/.local/lib/python3.8/site-packages (1.79)\n",
      "Requirement already satisfied: numpy in /home/maxim/.local/lib/python3.8/site-packages (from biopython) (1.21.2)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: elementpath in /home/maxim/.local/lib/python3.8/site-packages (3.0.2)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: pandas in /home/maxim/.local/lib/python3.8/site-packages (1.3.3)\n",
      "Requirement already satisfied: pytz>=2017.3 in /usr/lib/python3/dist-packages (from pandas) (2019.3)\n",
      "Requirement already satisfied: numpy>=1.17.3 in /home/maxim/.local/lib/python3.8/site-packages (from pandas) (1.21.2)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /home/maxim/.local/lib/python3.8/site-packages (from pandas) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in /home/maxim/.local/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas) (1.16.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: SQLAlchemy in /home/maxim/.local/lib/python3.8/site-packages (1.4.44)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /home/maxim/.local/lib/python3.8/site-packages (from SQLAlchemy) (2.0.1)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: db-sqlite3 in /home/maxim/.local/lib/python3.8/site-packages (0.0.1)\n",
      "Requirement already satisfied: db in /home/maxim/.local/lib/python3.8/site-packages (from db-sqlite3) (0.1.1)\n",
      "Requirement already satisfied: antiorm in /home/maxim/.local/lib/python3.8/site-packages (from db->db-sqlite3) (1.2.1)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: dnachisel in /home/maxim/.local/lib/python3.8/site-packages (3.2.10)\n",
      "Requirement already satisfied: Biopython in /home/maxim/.local/lib/python3.8/site-packages (from dnachisel) (1.79)\n",
      "Requirement already satisfied: python-codon-tables in /home/maxim/.local/lib/python3.8/site-packages (from dnachisel) (0.1.12)\n",
      "Requirement already satisfied: numpy in /home/maxim/.local/lib/python3.8/site-packages (from dnachisel) (1.21.2)\n",
      "Requirement already satisfied: docopt in /home/maxim/.local/lib/python3.8/site-packages (from dnachisel) (0.6.2)\n",
      "Requirement already satisfied: flametree in /home/maxim/.local/lib/python3.8/site-packages (from dnachisel) (0.1.12)\n",
      "Requirement already satisfied: proglog in /home/maxim/.local/lib/python3.8/site-packages (from dnachisel) (0.1.10)\n",
      "Requirement already satisfied: tqdm in /usr/lib/python3/dist-packages (from proglog->dnachisel) (4.30.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: pandasql in /home/maxim/.local/lib/python3.8/site-packages (0.7.3)\n",
      "Requirement already satisfied: numpy in /home/maxim/.local/lib/python3.8/site-packages (from pandasql) (1.21.2)\n",
      "Requirement already satisfied: pandas in /home/maxim/.local/lib/python3.8/site-packages (from pandasql) (1.3.3)\n",
      "Requirement already satisfied: sqlalchemy in /home/maxim/.local/lib/python3.8/site-packages (from pandasql) (1.4.44)\n",
      "Requirement already satisfied: pytz>=2017.3 in /usr/lib/python3/dist-packages (from pandas->pandasql) (2019.3)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /home/maxim/.local/lib/python3.8/site-packages (from pandas->pandasql) (2.8.2)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /home/maxim/.local/lib/python3.8/site-packages (from sqlalchemy->pandasql) (2.0.1)\n",
      "Requirement already satisfied: six>=1.5 in /home/maxim/.local/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas->pandasql) (1.16.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: taxoniq in /home/maxim/.local/lib/python3.8/site-packages (0.6.0)\n",
      "Requirement already satisfied: urllib3>=1.25.8 in /usr/lib/python3/dist-packages (from taxoniq) (1.25.8)\n",
      "Requirement already satisfied: ncbi-taxon-db>=2021.4.10 in /home/maxim/.local/lib/python3.8/site-packages (from taxoniq) (2021.11.16)\n",
      "Requirement already satisfied: zstandard>=0.15.1 in /home/maxim/.local/lib/python3.8/site-packages (from taxoniq) (0.19.0)\n",
      "Requirement already satisfied: ncbi-refseq-accession-lengths==2021.11.16 in /home/maxim/.local/lib/python3.8/site-packages (from ncbi-taxon-db>=2021.4.10->taxoniq) (2021.11.16)\n",
      "Requirement already satisfied: ncbi-refseq-accession-offsets==2021.11.16 in /home/maxim/.local/lib/python3.8/site-packages (from ncbi-taxon-db>=2021.4.10->taxoniq) (2021.11.16)\n",
      "Requirement already satisfied: ncbi-refseq-accession-db==2021.11.16 in /home/maxim/.local/lib/python3.8/site-packages (from ncbi-taxon-db>=2021.4.10->taxoniq) (2021.11.16)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: nglview in /home/maxim/.local/lib/python3.8/site-packages (3.0.3)\n",
      "Requirement already satisfied: numpy in /home/maxim/.local/lib/python3.8/site-packages (from nglview) (1.21.2)\n",
      "Requirement already satisfied: jupyterlab-widgets in /home/maxim/.local/lib/python3.8/site-packages (from nglview) (3.0.3)\n",
      "Requirement already satisfied: ipywidgets>=7 in /home/maxim/.local/lib/python3.8/site-packages (from nglview) (7.0.0)\n",
      "Requirement already satisfied: widgetsnbextension~=3.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from ipywidgets>=7->nglview) (3.0.8)\n",
      "Requirement already satisfied: nbformat>=4.2.0 in /home/maxim/.local/lib/python3.8/site-packages (from ipywidgets>=7->nglview) (5.7.0)\n",
      "Requirement already satisfied: traitlets>=4.3.1 in /home/maxim/.local/lib/python3.8/site-packages (from ipywidgets>=7->nglview) (5.5.0)\n",
      "Requirement already satisfied: ipython>=4.0.0 in /usr/lib/python3/dist-packages (from ipywidgets>=7->nglview) (7.13.0)\n",
      "Requirement already satisfied: ipykernel>=4.5.1 in /usr/lib/python3/dist-packages (from ipywidgets>=7->nglview) (5.2.0)\n",
      "Requirement already satisfied: pexpect in /usr/lib/python3/dist-packages (from ipython>=4.0.0->ipywidgets>=7->nglview) (4.6.0)\n",
      "Requirement already satisfied: jupyter-core in /home/maxim/.local/lib/python3.8/site-packages (from nbformat>=4.2.0->ipywidgets>=7->nglview) (5.0.0)\n",
      "Requirement already satisfied: jsonschema>=2.6 in /home/maxim/.local/lib/python3.8/site-packages (from nbformat>=4.2.0->ipywidgets>=7->nglview) (3.2.0)\n",
      "Requirement already satisfied: fastjsonschema in /home/maxim/.local/lib/python3.8/site-packages (from nbformat>=4.2.0->ipywidgets>=7->nglview) (2.16.2)\n",
      "Requirement already satisfied: notebook>=4.4.1 in /home/maxim/.local/lib/python3.8/site-packages (from widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (6.5.2)\n",
      "Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (from jsonschema>=2.6->nbformat>=4.2.0->ipywidgets>=7->nglview) (45.2.0)\n",
      "Requirement already satisfied: pyrsistent>=0.14.0 in /home/maxim/.local/lib/python3.8/site-packages (from jsonschema>=2.6->nbformat>=4.2.0->ipywidgets>=7->nglview) (0.16.1)\n",
      "Requirement already satisfied: six>=1.11.0 in /home/maxim/.local/lib/python3.8/site-packages (from jsonschema>=2.6->nbformat>=4.2.0->ipywidgets>=7->nglview) (1.16.0)\n",
      "Requirement already satisfied: attrs>=17.4.0 in /home/maxim/.local/lib/python3.8/site-packages (from jsonschema>=2.6->nbformat>=4.2.0->ipywidgets>=7->nglview) (22.2.0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: jinja2 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (3.1.2)\n",
      "Requirement already satisfied: jupyter-client>=5.3.4 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (7.4.7)\n",
      "Requirement already satisfied: Send2Trash>=1.8.0 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.8.0)\n",
      "Requirement already satisfied: terminado>=0.8.3 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.17.0)\n",
      "Requirement already satisfied: tornado>=6.1 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (6.2)\n",
      "Requirement already satisfied: prometheus-client in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.15.0)\n",
      "Requirement already satisfied: nest-asyncio>=1.5 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.5.6)\n",
      "Requirement already satisfied: nbclassic>=0.4.7 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.4.8)\n",
      "Requirement already satisfied: pyzmq>=17 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (24.0.1)\n",
      "Requirement already satisfied: argon2-cffi in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (21.3.0)\n",
      "Requirement already satisfied: ipython-genutils in /usr/lib/python3/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.2.0)\n",
      "Requirement already satisfied: nbconvert>=5 in /home/maxim/.local/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (7.2.5)\n",
      "Requirement already satisfied: platformdirs in /usr/local/lib/python3.8/dist-packages (from jupyter-core->nbformat>=4.2.0->ipywidgets>=7->nglview) (2.5.4)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /home/maxim/.local/lib/python3.8/site-packages (from jupyter-client>=5.3.4->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.8.2)\n",
      "Requirement already satisfied: entrypoints in /usr/lib/python3/dist-packages (from jupyter-client>=5.3.4->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.3)\n",
      "Requirement already satisfied: notebook-shim>=0.1.0 in /home/maxim/.local/lib/python3.8/site-packages (from nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.2.2)\n",
      "Requirement already satisfied: jupyter-server>=1.8 in /home/maxim/.local/lib/python3.8/site-packages (from nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.23.2)\n",
      "Requirement already satisfied: markupsafe>=2.0 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.1.1)\n",
      "Requirement already satisfied: mistune<3,>=2.0.3 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.0.4)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in /usr/lib/python3/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.4.2)\n",
      "Requirement already satisfied: tinycss2 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.2.1)\n",
      "Requirement already satisfied: beautifulsoup4 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (4.5.3)\n",
      "Requirement already satisfied: pygments>=2.4.1 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.13.0)\n",
      "Requirement already satisfied: importlib-metadata>=3.6 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (5.0.0)\n",
      "Requirement already satisfied: defusedxml in /usr/lib/python3/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.6.0)\n",
      "Requirement already satisfied: packaging in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (21.3)\n",
      "Requirement already satisfied: bleach in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (5.0.1)\n",
      "Requirement already satisfied: nbclient>=0.5.0 in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.7.0)\n",
      "Requirement already satisfied: jupyterlab-pygments in /home/maxim/.local/lib/python3.8/site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.2.2)\n",
      "Requirement already satisfied: ptyprocess in /home/maxim/.local/lib/python3.8/site-packages (from terminado>=0.8.3->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.7.0)\n",
      "Requirement already satisfied: argon2-cffi-bindings in /home/maxim/.local/lib/python3.8/site-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (21.2.0)\n",
      "Requirement already satisfied: zipp>=0.5 in /home/maxim/.local/lib/python3.8/site-packages (from importlib-metadata>=3.6->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (3.10.0)\n",
      "Requirement already satisfied: anyio<4,>=3.1.0 in /home/maxim/.local/lib/python3.8/site-packages (from jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (3.6.2)\n",
      "Requirement already satisfied: websocket-client in /home/maxim/.local/lib/python3.8/site-packages (from jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.4.2)\n",
      "Requirement already satisfied: cffi>=1.0.1 in /home/maxim/.local/lib/python3.8/site-packages (from argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.15.1)\n",
      "Requirement already satisfied: webencodings in /usr/lib/python3/dist-packages (from bleach->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (0.5.1)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/lib/python3/dist-packages (from packaging->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.4.6)\n",
      "Requirement already satisfied: sniffio>=1.1 in /home/maxim/.local/lib/python3.8/site-packages (from anyio<4,>=3.1.0->jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (1.3.0)\n",
      "Requirement already satisfied: idna>=2.8 in /usr/lib/python3/dist-packages (from anyio<4,>=3.1.0->jupyter-server>=1.8->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.8)\n",
      "Requirement already satisfied: pycparser in /home/maxim/.local/lib/python3.8/site-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.0.0->ipywidgets>=7->nglview) (2.21)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: biopandas in /home/maxim/.local/lib/python3.8/site-packages (0.4.1)\n",
      "Requirement already satisfied: numpy>=1.16.2 in /home/maxim/.local/lib/python3.8/site-packages (from biopandas) (1.21.2)\n",
      "Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (from biopandas) (45.2.0)\n",
      "Requirement already satisfied: pandas>=0.24.2 in /home/maxim/.local/lib/python3.8/site-packages (from biopandas) (1.3.3)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /home/maxim/.local/lib/python3.8/site-packages (from pandas>=0.24.2->biopandas) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2017.3 in /usr/lib/python3/dist-packages (from pandas>=0.24.2->biopandas) (2019.3)\n",
      "Requirement already satisfied: six>=1.5 in /home/maxim/.local/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas>=0.24.2->biopandas) (1.16.0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: tabulate in /home/maxim/.local/lib/python3.8/site-packages (0.9.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: collection in /home/maxim/.local/lib/python3.8/site-packages (0.1.6)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: contextlib2 in /home/maxim/.local/lib/python3.8/site-packages (21.6.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: fuzzywuzzy in /home/maxim/.local/lib/python3.8/site-packages (0.18.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: pyhmmer in /home/maxim/.local/lib/python3.8/site-packages (0.7.0)\n",
      "Requirement already satisfied: psutil~=5.8 in /home/maxim/.local/lib/python3.8/site-packages (from pyhmmer) (5.9.4)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: sto in /home/maxim/.local/lib/python3.8/site-packages (0.4.1)\n",
      "Requirement already satisfied: click in /home/maxim/.local/lib/python3.8/site-packages (from sto) (8.1.3)\n",
      "Requirement already satisfied: configobj in /home/maxim/.local/lib/python3.8/site-packages (from sto) (5.0.6)\n",
      "Requirement already satisfied: eth-abi<1.3 in /home/maxim/.local/lib/python3.8/site-packages (from sto) (1.2.2)\n",
      "Requirement already satisfied: colorama in /usr/lib/python3/dist-packages (from sto) (0.4.3)\n",
      "Requirement already satisfied: coloredlogs in /usr/lib/python3/dist-packages (from sto) (7.3)\n",
      "Requirement already satisfied: sqlalchemy>=1.3.0b1 in /home/maxim/.local/lib/python3.8/site-packages (from sto) (1.4.44)\n",
      "Requirement already satisfied: web3[tester] in /home/maxim/.local/lib/python3.8/site-packages (from sto) (4.9.1)\n",
      "Requirement already satisfied: eth-typing<2 in /home/maxim/.local/lib/python3.8/site-packages (from sto) (1.3.0)\n",
      "Requirement already satisfied: tabulate in /home/maxim/.local/lib/python3.8/site-packages (from sto) (0.9.0)\n",
      "Requirement already satisfied: tqdm in /usr/lib/python3/dist-packages (from sto) (4.30.0)\n",
      "Requirement already satisfied: arrow in /home/maxim/.local/lib/python3.8/site-packages (from sto) (1.2.3)\n",
      "Requirement already satisfied: pysha3<2.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from sto) (1.0.2)\n",
      "Requirement already satisfied: eth-utils<2.0.0,>=1.2.0 in /home/maxim/.local/lib/python3.8/site-packages (from eth-abi<1.3->sto) (1.8.1)\n",
      "Requirement already satisfied: parsimonious<0.9.0,>=0.8.0 in /home/maxim/.local/lib/python3.8/site-packages (from eth-abi<1.3->sto) (0.8.1)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /home/maxim/.local/lib/python3.8/site-packages (from sqlalchemy>=1.3.0b1->sto) (2.0.1)\n",
      "Requirement already satisfied: python-dateutil>=2.7.0 in /home/maxim/.local/lib/python3.8/site-packages (from arrow->sto) (2.8.2)\n",
      "Requirement already satisfied: six in /home/maxim/.local/lib/python3.8/site-packages (from configobj->sto) (1.16.0)\n",
      "Requirement already satisfied: eth-hash[pycryptodome]<1.0.0,>=0.2.0 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (0.3.3)\n",
      "Requirement already satisfied: cytoolz<1.0.0,>=0.9.0 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (0.12.0)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.16.0 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (2.28.1)\n",
      "Requirement already satisfied: lru-dict<2.0.0,>=1.1.6 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (1.1.8)\n",
      "Requirement already satisfied: eth-account<0.4.0,>=0.2.1 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (0.3.0)\n",
      "Requirement already satisfied: hexbytes<1.0.0,>=0.1.0 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (0.3.0)\n",
      "Requirement already satisfied: websockets<7.0.0,>=6.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (6.0)\n",
      "Requirement already satisfied: py-geth<3.0.0,>=2.0.1 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (2.4.0)\n",
      "Requirement already satisfied: eth-tester[py-evm]==0.1.0-beta.33 in /home/maxim/.local/lib/python3.8/site-packages (from web3[tester]->sto) (0.1.0b33)\n",
      "Requirement already satisfied: rlp<2.0.0,>=0.6.0 in /home/maxim/.local/lib/python3.8/site-packages (from eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (1.2.0)\n",
      "Requirement already satisfied: semantic-version<3.0.0,>=2.6.0 in /home/maxim/.local/lib/python3.8/site-packages (from eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (2.10.0)\n",
      "Requirement already satisfied: eth-keys<0.3.0,>=0.2.0-beta.3 in /home/maxim/.local/lib/python3.8/site-packages (from eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (0.2.4)\n",
      "Requirement already satisfied: py-evm==0.2.0a33 in /home/maxim/.local/lib/python3.8/site-packages (from eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (0.2.0a33)\n",
      "Requirement already satisfied: py-ecc<2.0.0,>=1.4.2 in /home/maxim/.local/lib/python3.8/site-packages (from py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (1.4.7)\n",
      "Requirement already satisfied: pyethash<1.0.0,>=0.1.27 in /home/maxim/.local/lib/python3.8/site-packages (from py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (0.1.27)\n",
      "Requirement already satisfied: trie<2.0.0,>=1.3.5 in /home/maxim/.local/lib/python3.8/site-packages (from py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (1.4.0)\n",
      "Requirement already satisfied: cryptography<3.0.0,>=2.0.3 in /home/maxim/.local/lib/python3.8/site-packages (from py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (2.9.2)\n",
      "Requirement already satisfied: eth-bloom<2.0.0,>=1.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (1.0.4)\n",
      "Requirement already satisfied: toolz>=0.8.0 in /usr/lib/python3/dist-packages (from cytoolz<1.0.0,>=0.9.0->web3[tester]->sto) (0.9.0)\n",
      "Requirement already satisfied: attrdict<3,>=2.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from eth-account<0.4.0,>=0.2.1->web3[tester]->sto) (2.0.1)\n",
      "Requirement already satisfied: eth-rlp<1,>=0.1.2 in /home/maxim/.local/lib/python3.8/site-packages (from eth-account<0.4.0,>=0.2.1->web3[tester]->sto) (0.2.1)\n",
      "Requirement already satisfied: eth-keyfile<0.6.0,>=0.5.0 in /home/maxim/.local/lib/python3.8/site-packages (from eth-account<0.4.0,>=0.2.1->web3[tester]->sto) (0.5.1)\n",
      "Requirement already satisfied: pycryptodome<4,>=3.6.6 in /home/maxim/.local/lib/python3.8/site-packages (from eth-hash[pycryptodome]<1.0.0,>=0.2.0->web3[tester]->sto) (3.16.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests<3.0.0,>=2.16.0->web3[tester]->sto) (2.8)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/lib/python3/dist-packages (from requests<3.0.0,>=2.16.0->web3[tester]->sto) (1.25.8)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /home/maxim/.local/lib/python3.8/site-packages (from requests<3.0.0,>=2.16.0->web3[tester]->sto) (2.1.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests<3.0.0,>=2.16.0->web3[tester]->sto) (2019.11.28)\n",
      "Requirement already satisfied: cffi!=1.11.3,>=1.8 in /home/maxim/.local/lib/python3.8/site-packages (from cryptography<3.0.0,>=2.0.3->py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (1.15.1)\n",
      "Requirement already satisfied: pycparser in /home/maxim/.local/lib/python3.8/site-packages (from cffi!=1.11.3,>=1.8->cryptography<3.0.0,>=2.0.3->py-evm==0.2.0a33->eth-tester[py-evm]==0.1.0-beta.33->web3[tester]->sto) (2.21)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: bio in /home/maxim/.local/lib/python3.8/site-packages (1.5.2)\n",
      "Requirement already satisfied: requests in /home/maxim/.local/lib/python3.8/site-packages (from bio) (2.28.1)\n",
      "Requirement already satisfied: mygene in /home/maxim/.local/lib/python3.8/site-packages (from bio) (3.2.2)\n",
      "Requirement already satisfied: tqdm in /usr/lib/python3/dist-packages (from bio) (4.30.0)\n",
      "Requirement already satisfied: biopython==1.79 in /home/maxim/.local/lib/python3.8/site-packages (from bio) (1.79)\n",
      "Requirement already satisfied: numpy in /home/maxim/.local/lib/python3.8/site-packages (from biopython==1.79->bio) (1.21.2)\n",
      "Requirement already satisfied: biothings-client>=0.2.6 in /home/maxim/.local/lib/python3.8/site-packages (from mygene->bio) (0.2.6)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests->bio) (2.8)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /home/maxim/.local/lib/python3.8/site-packages (from requests->bio) (2.1.1)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/lib/python3/dist-packages (from requests->bio) (1.25.8)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests->bio) (2019.11.28)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: hmmlearn in /home/maxim/.local/lib/python3.8/site-packages (0.2.8)\n",
      "Requirement already satisfied: scipy>=0.19 in /usr/lib/python3/dist-packages (from hmmlearn) (1.3.3)\n",
      "Requirement already satisfied: scikit-learn>=0.16 in /home/maxim/.local/lib/python3.8/site-packages (from hmmlearn) (1.1.1)\n",
      "Requirement already satisfied: numpy>=1.10 in /home/maxim/.local/lib/python3.8/site-packages (from hmmlearn) (1.21.2)\n",
      "Requirement already satisfied: joblib>=1.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from scikit-learn>=0.16->hmmlearn) (1.1.0)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /home/maxim/.local/lib/python3.8/site-packages (from scikit-learn>=0.16->hmmlearn) (3.1.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: unipressed in /home/maxim/.local/lib/python3.8/site-packages (1.1.0)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.28.1 in /home/maxim/.local/lib/python3.8/site-packages (from unipressed) (2.28.1)\n",
      "Requirement already satisfied: typing-extensions<5.0.0,>=4.3.0 in /usr/local/lib/python3.8/dist-packages (from unipressed) (4.4.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests<3.0.0,>=2.28.1->unipressed) (2.8)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /home/maxim/.local/lib/python3.8/site-packages (from requests<3.0.0,>=2.28.1->unipressed) (2.1.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests<3.0.0,>=2.28.1->unipressed) (2019.11.28)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/lib/python3/dist-packages (from requests<3.0.0,>=2.28.1->unipressed) (1.25.8)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: uniprot in /home/maxim/.local/lib/python3.8/site-packages (1.3)\n",
      "Requirement already satisfied: requests in /home/maxim/.local/lib/python3.8/site-packages (from uniprot) (2.28.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests->uniprot) (2019.11.28)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/lib/python3/dist-packages (from requests->uniprot) (1.25.8)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests->uniprot) (2.8)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /home/maxim/.local/lib/python3.8/site-packages (from requests->uniprot) (2.1.1)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: requests in /home/maxim/.local/lib/python3.8/site-packages (2.28.1)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/lib/python3/dist-packages (from requests) (1.25.8)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests) (2019.11.28)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /home/maxim/.local/lib/python3.8/site-packages (from requests) (2.1.1)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests) (2.8)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "\u001b[31mERROR: Could not find a version that satisfies the requirement subprocess (from versions: none)\u001b[0m\u001b[31m\n",
      "\u001b[0m\u001b[31mERROR: No matching distribution found for subprocess\u001b[0m\u001b[31m\n",
      "\u001b[0mDefaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: dict2xml in /home/maxim/.local/lib/python3.8/site-packages (1.7.2)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: minidom-ext in /home/maxim/.local/lib/python3.8/site-packages (0.0.6)\n",
      "Requirement already satisfied: lxml>=4.5.1 in /home/maxim/.local/lib/python3.8/site-packages (from minidom-ext) (4.9.2)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: flatten-json in /home/maxim/.local/lib/python3.8/site-packages (0.1.13)\n",
      "Requirement already satisfied: six in /home/maxim/.local/lib/python3.8/site-packages (from flatten-json) (1.16.0)\n",
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Requirement already satisfied: cdifflib in /home/maxim/.local/lib/python3.8/site-packages (1.2.6)\n"
     ]
    }
   ],
   "source": [
    "# Install needed packages\n",
    "# biopython for biological analitical methods\n",
    "# elementpath for tree itiration\n",
    "# pandas for the construction of a dataframe from the xml file\n",
    "# SQLAlchemy and db-sqlite3 for database parsing from the dataframe\n",
    "!pip install biopython\n",
    "!pip install elementpath\n",
    "!pip install pandas\n",
    "!pip install SQLAlchemy\n",
    "!pip install db-sqlite3\n",
    "!pip install pandasql\n",
    "!pip3 install taxoniq\n",
    "!pip install nglview\n",
    "!pip install biopandas\n",
    "!pip install tabulate\n",
    "!pip install collection\n",
    "!pip install contextlib2\n",
    "!pip install fuzzywuzzy\n",
    "!pip install pyhmmer\n",
    "!pip install sto\n",
    "!pip install bio\n",
    "!pip install hmmlearn\n",
    "!pip install unipressed\n",
    "!pip install uniprot\n",
    "!pip install requests\n",
    "!pip install subprocess\n",
    "!pip install dict2xml \n",
    "!pip install minidom-ext\n",
    "!pip install flatten-json\n",
    "!pip install cdifflib\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "7e3e9c1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# query protein sequence input\n",
    "from Bio import SeqIO\n",
    "protein_seq = SeqIO.read(\"sequence.842.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "26304330",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFKVLLCGA...KHW')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# double check the sequence\n",
    "protein_seq.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "534ffb5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run an online blast search in the NCBI database\n",
    "from Bio.Blast import NCBIWWW\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"pdb\", protein_seq.seq) # pdb - protein data bank"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bcceb543",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the ncbi search results as an xml document\n",
    "with open(\"TEM-1.xml\", \"w\") as save_to:\n",
    "    save_to.write(result_handle.read())\n",
    "    result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "299610d7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Protein bank ID accession Length  \\\n",
      "0       pdb|7QLP|A    7QLP_A    263   \n",
      "1       pdb|1ERO|A    1ERO_A    263   \n",
      "2       pdb|5NPO|A    5NPO_A    264   \n",
      "3       pdb|1ESU|A    1ESU_A    263   \n",
      "4       pdb|1JTD|A    1JTD_A    263   \n",
      "5       pdb|1LI9|A    1LI9_A    263   \n",
      "6       pdb|1JWV|A    1JWV_A    263   \n",
      "7       pdb|1ZG4|A    1ZG4_A    286   \n",
      "8       pdb|1FQG|A    1FQG_A    263   \n",
      "9       pdb|1LHY|A    1LHY_A    263   \n",
      "10      pdb|1AXB|A    1AXB_A    263   \n",
      "11      pdb|1YT4|A    1YT4_A    263   \n",
      "12      pdb|1PZP|A    1PZP_A    263   \n",
      "13      pdb|1JVJ|A    1JVJ_A    263   \n",
      "14      pdb|1JWP|A    1JWP_A    263   \n",
      "15      pdb|6B2N|A    6B2N_A    263   \n",
      "16      pdb|7U6Q|A    7U6Q_A    264   \n",
      "17      pdb|3JYI|A    3JYI_A    263   \n",
      "18      pdb|1ERM|A    1ERM_A    263   \n",
      "19      pdb|5KKF|A    5KKF_A    263   \n",
      "20      pdb|3CMZ|A    3CMZ_A    263   \n",
      "21      pdb|1LI0|A    1LI0_A    263   \n",
      "22      pdb|4MEZ|A    4MEZ_A    263   \n",
      "23      pdb|1ZG6|A    1ZG6_A    286   \n",
      "24      pdb|4ZJ1|A    4ZJ1_A    297   \n",
      "25      pdb|1CK3|A    1CK3_A    263   \n",
      "26      pdb|6APA|A    6APA_A    263   \n",
      "27      pdb|8DDZ|A    8DDZ_A    264   \n",
      "28      pdb|1JWZ|A    1JWZ_A    263   \n",
      "29      pdb|1HTZ|A    1HTZ_A    263   \n",
      "30      pdb|5I52|A    5I52_A    263   \n",
      "31      pdb|4ZJ2|A    4ZJ2_A    297   \n",
      "32      pdb|1XXM|A    1XXM_A    263   \n",
      "33      pdb|2B5R|A    2B5R_A    263   \n",
      "34      pdb|5IQ8|A    5IQ8_A    263   \n",
      "35      pdb|3GMW|A    3GMW_A    261   \n",
      "36      pdb|3P98|A    3P98_A    286   \n",
      "37      pdb|4ZJ3|A    4ZJ3_A    297   \n",
      "38      pdb|4RX3|A    4RX3_A    263   \n",
      "39      pdb|4RX2|A    4RX2_A    263   \n",
      "40      pdb|4RVA|A    4RVA_A    263   \n",
      "41      pdb|4IBX|A    4IBX_A    263   \n",
      "42      pdb|4OP5|A    4OP5_A    263   \n",
      "43      pdb|4OP8|A    4OP8_A    263   \n",
      "44      pdb|5NPO|B    5NPO_B    265   \n",
      "45      pdb|4IBR|A    4IBR_A    263   \n",
      "46      pdb|3DTM|A    3DTM_A    263   \n",
      "47      pdb|4OPQ|A    4OPQ_A    263   \n",
      "48      pdb|2V1Z|A    2V1Z_A    291   \n",
      "49      pdb|4ID4|A    4ID4_A    263   \n",
      "\n",
      "                                            Query seq  \\\n",
      "0   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "1   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "2   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "3   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "4   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "5   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "6   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "7   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "8   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "9   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "10  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "11  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "12  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "13  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "14  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "15  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "16  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "17  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "18  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "19  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "20  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "21  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "22  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "23  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "24  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "25  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "26  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "27  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "28  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "29  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "30  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "31  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "32  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "33  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "34  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "35  ETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFK...   \n",
      "36  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "37  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "38  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "39  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "40  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "41  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "42  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "43  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "44  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "45  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "46  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "47  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "48  PETLVKVKDAEDQLG------ARVGYIELDLNSGKILESFRPEERF...   \n",
      "49  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...   \n",
      "\n",
      "                                            Match seq  \n",
      "0   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "1   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "2   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "3   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "4   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "5   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPM+ST...  \n",
      "6   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "7   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "8   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "9   HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "10  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "11  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "12  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "13  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "14  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "15  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "16  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "17  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "18  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "19  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "20  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "21  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPM+ST...  \n",
      "22  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFP+ ST...  \n",
      "23  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMM T...  \n",
      "24  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "25  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "26  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "27  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "28  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "29  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "30  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "31  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMM T...  \n",
      "32  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "33  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "34  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "35  ETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFK...  \n",
      "36  HPETLVKVKDAED+LGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "37  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMM T...  \n",
      "38  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMM T...  \n",
      "39  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "40  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n",
      "41  HPETLVKVKDAEDQLG RVGYIELDL SGKILESFRPEERFPMMST...  \n",
      "42  HPETLVKVKDAEDQLG RVGYIELDL SGKILESFRPEERFPMMST...  \n",
      "43  HPETLVKVKDAEDQLG RVGYIELDL SGKILESFRPEERFPMMST...  \n",
      "44  HPETLVKVKDA+DQLG RVGYIELDL SGKIL SFRPEERFPMMST...  \n",
      "45  HPETLVKVKDAEDQLG RVGYIELDL SGKILESFRPEERFPMMST...  \n",
      "46  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFR EERFPMMST...  \n",
      "47  HPETLVKVKDAEDQLG RVGYIELDL SGKILESFRPEERFPMMST...  \n",
      "48  PETLVKVKDAEDQL        RVGYIELDLNSGKILESFRPEERF...  \n",
      "49  HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMST...  \n"
     ]
    }
   ],
   "source": [
    "# convert the xml to a dataframe\n",
    "\n",
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"Protein bank ID\",\"accession\",\"Length\",\"Query seq\",\"Match seq\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "    if child.tag == 'Hit_id':\n",
    "        Hit_id=child.text.strip()\n",
    "    elif child.tag == 'Hit_accession':\n",
    "        Hit_accession=child.text.strip()\n",
    "    elif child.tag == 'Hit_len':\n",
    "        Hit_len=child.text.strip()\n",
    "    elif child.tag == 'Hsp_qseq':\n",
    "        Hsp_qseq=child.text.strip()\n",
    "    elif child.tag == 'Hsp_midline':\n",
    "        Hsp_midline=child.text.strip()\n",
    "# append data of interest into the database        \n",
    "        rows.append({\"Protein bank ID\": Hit_id,\"accession\": Hit_accession,\"Length\": Hit_len,\"Query seq\": Hsp_qseq,\n",
    "                     \"Match seq\": Hsp_midline})\n",
    "# create a dataframe and visualize it\n",
    "df = pd.DataFrame(rows, columns=cols)\n",
    "print(df) \n",
    "# Writing dataframe to csv\n",
    "#df.to_csv('my_blast1.csv',header=None,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "6915b42c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7QLP_A\n",
      "1ERO_A\n",
      "5NPO_A\n",
      "1ESU_A\n",
      "1JTD_A\n",
      "1LI9_A\n",
      "1JWV_A\n",
      "1ZG4_A\n",
      "1FQG_A\n",
      "1LHY_A\n",
      "1AXB_A\n",
      "1YT4_A\n",
      "1PZP_A\n",
      "1JVJ_A\n",
      "1JWP_A\n",
      "6B2N_A\n",
      "7U6Q_A\n",
      "3JYI_A\n",
      "1ERM_A\n",
      "5KKF_A\n",
      "3CMZ_A\n",
      "1LI0_A\n",
      "4MEZ_A\n",
      "1ZG6_A\n",
      "4ZJ1_A\n",
      "1CK3_A\n",
      "6APA_A\n",
      "8DDZ_A\n",
      "1JWZ_A\n",
      "1HTZ_A\n",
      "5I52_A\n",
      "4ZJ2_A\n",
      "1XXM_A\n",
      "2B5R_A\n",
      "5IQ8_A\n",
      "3GMW_A\n",
      "3P98_A\n",
      "4ZJ3_A\n",
      "4RX3_A\n",
      "4RX2_A\n",
      "4RVA_A\n",
      "4IBX_A\n",
      "4OP5_A\n",
      "4OP8_A\n",
      "5NPO_B\n",
      "4IBR_A\n",
      "3DTM_A\n",
      "4OPQ_A\n",
      "2V1Z_A\n",
      "4ID4_A\n"
     ]
    }
   ],
   "source": [
    "from Bio.Blast import NCBIXML\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'maximschwarzberg@gmail.com'\n",
    "#print(df[\"accession\"])\n",
    "df2 = df[\"accession\"]\n",
    "#print(df2)\n",
    "index = df2.reset_index()\n",
    "for index, row in df.iterrows():\n",
    "    #extracts all of the accession numbers\n",
    "    print(row[\"accession\"])\n",
    "    response = Entrez.efetch(db=\"protein\", id=row[\"accession\"], retmode=\"xml\")\n",
    "    result = list(Entrez.parse(response))\n",
    "    #here you can look at all of the availble data which comes from each of the BLAST \n",
    "    #sequences which were found\n",
    "    #print(result[:1])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0f28e353",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez, SeqIO\n",
    "from Bio.Blast import NCBIXML\n",
    "from Bio import Entrez\n",
    "import taxoniq\n",
    "\n",
    "Entrez.email = 'maximschwarzberg@gmail.com'\n",
    "#print(df[\"accession\"])\n",
    "df2 = df[\"accession\"]\n",
    "#print(df2)\n",
    "index = df2.reset_index()\n",
    "for index, row in df.iterrows():\n",
    "    #print(row[\"accession\"])\n",
    "    response = Entrez.efetch(db=\"protein\", id=row[\"accession\"], retmode=\"xml\")\n",
    "    result = list(Entrez.parse(response))\n",
    "    t2 = taxoniq.Taxon(scientific_name=(result[0][\"GBSeq_organism\"]))\n",
    "    #print(result[0][\"GBSeq_organism\"] ,t2.tax_id)\n",
    "    with open(\"name.txt\", \"a\") as f:\n",
    "        print(result[0][\"GBSeq_organism\"] , file=f)\n",
    "    with open(\"taxo.txt\", \"a\") as f:\n",
    "        print(t2.tax_id, file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2e2a8bb5",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7QLP 2.3\n",
      "1ERO 2.1\n",
      "5NPO 1.95\n",
      "1ESU 2.0\n",
      "1JTD 2.3\n",
      "1LI9 1.52\n",
      "1JWV 1.85\n",
      "1ZG4 1.55\n",
      "1FQG 1.7\n",
      "1LHY 2.0\n",
      "1AXB 2.0\n",
      "1YT4 1.4\n",
      "1PZP 1.45\n",
      "1JVJ 1.73\n",
      "1JWP 1.75\n",
      "6B2N 2.0\n",
      "7U6Q 1.9\n",
      "3JYI 2.7\n",
      "1ERM 1.7\n",
      "5KKF 1.82\n",
      "3CMZ 1.92\n",
      "1LI0 1.61\n",
      "4MEZ 2.05\n",
      "1ZG6 2.1\n",
      "4ZJ1 1.54\n",
      "1CK3 2.28\n",
      "6APA 1.86\n",
      "8DDZ 1.45\n",
      "1JWZ 1.8\n",
      "1HTZ 2.4\n",
      "5I52 1.75\n",
      "4ZJ2 1.8\n",
      "1XXM 1.9\n",
      "2B5R 1.65\n",
      "5IQ8 2.06\n",
      "3GMW 2.1\n",
      "3P98 2.1\n",
      "4ZJ3 1.7\n",
      "4RX3 1.39\n",
      "4RX2 2.31\n",
      "4RVA 1.44\n",
      "4IBX 2.68\n",
      "4OP5 1.05\n",
      "4OP8 1.95\n",
      "5NPO 1.95\n",
      "4IBR 2.2\n",
      "3DTM 2.0\n",
      "4OPQ 1.7\n",
      "2V1Z 1.6\n",
      "4ID4 1.05\n"
     ]
    }
   ],
   "source": [
    "from Bio.PDB import *\n",
    "from tempfile import gettempdir\n",
    "import biotite.database.rcsb as rcsb\n",
    "from Bio.PDB import MMCIFParser\n",
    "from Bio.PDB.PDBExceptions import PDBConstructionWarning\n",
    "import re\n",
    "import warnings\n",
    "import pandas as pd\n",
    "import sys\n",
    "\n",
    "#cleaning the data from e.g 4ID4_A to 4ID4 for each of the PDB ids\n",
    "df3 = df[\"accession\"].apply(lambda x: re.findall(r'^.{4}', x)[0])\n",
    "#print(df3)\n",
    "df4 = pd.DataFrame(df3, columns=['accession'])\n",
    "#print(df4)\n",
    "index = df4.reset_index()\n",
    "for index, row in df4.iterrows():\n",
    "    file_path = rcsb.fetch(row[\"accession\"], \"pdb\", gettempdir())\n",
    "    warnings.simplefilter('ignore', PDBConstructionWarning)\n",
    "    parser = PDBParser()\n",
    "    structure  = parser.get_structure(\"?\", file_path)\n",
    "    #extract PDB id next to its resolution                                 \n",
    "    pdb_code = structure.header.get(\"idcode\")\n",
    "    resolution = structure.header.get(\"resolution\")\n",
    "    print(pdb_code ,resolution)\n",
    "    #save as txt for index reset\n",
    "    with open(\"output.txt\", \"a\") as f:\n",
    "        print(pdb_code, resolution, file=f)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0c1a3269",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   pdb_code  resolution\n",
      "42     4OP5        1.05\n",
      "49     4ID4        1.05\n",
      "92     4OP5        1.05\n",
      "99     4ID4        1.05\n",
      "38     4RX3        1.39\n",
      "88     4RX3        1.39\n",
      "11     1YT4        1.40\n",
      "61     1YT4        1.40\n",
      "40     4RVA        1.44\n",
      "90     4RVA        1.44\n",
      "  pdb_code\n",
      "0     4OP5\n",
      "1     4ID4\n",
      "2     4OP5\n",
      "3     4ID4\n",
      "4     4RX3\n",
      "5     4RX3\n",
      "6     1YT4\n",
      "7     1YT4\n",
      "8     4RVA\n",
      "9     4RVA\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "#open the txt file from the above cell\n",
    "data = pd.read_csv('output.txt', sep=' ',header=None)\n",
    "#create a dataframe\n",
    "data.columns = [\"pdb_code\", \"resolution\", ]\n",
    "#extract the 10 smallest resolutions\n",
    "datamin = data.nsmallest(10, ['resolution'])\n",
    "#extract the 10 smallest PDB ids which correlate to the resolution above \n",
    "dataminparser = datamin['pdb_code']\n",
    "#create such a column\n",
    "dataminparser.columns = [\"pdb_code\"]\n",
    "#reset index\n",
    "dataminparser = dataminparser.reset_index(level=0)\n",
    "#create a new column\n",
    "dataminparser=  dataminparser[\"pdb_code\"]\n",
    "#create a dataframe with 10 smallest PDBs\n",
    "minPDB=pd.DataFrame(dataminparser,columns=['pdb_code'])\n",
    "#with resolution\n",
    "print(datamin)\n",
    "#only 10 ids\n",
    "print(minPDB)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "381de510",
   "metadata": {},
   "outputs": [],
   "source": [
    "from unipressed import UniprotkbClient\n",
    "import requests\n",
    "#extracting data from the uniprot database with uniprot id\n",
    "# Set the base URL for the UniProtKB REST API\n",
    "base_url = \"https://www.uniprot.org/uniprot/\"\n",
    "\n",
    "# Set the accession number of the protein you want to search for\n",
    "accession = \"Q6SJ61\"\n",
    "\n",
    "# Send a GET request to the UniProtKB REST API to retrieve the annotation data for the protein\n",
    "response = requests.get(base_url + accession + \".xml\")\n",
    "\n",
    "# Print the response\n",
    "#print(response.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b971b9b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/tmp/1LI9_A.fa', '/tmp/1ZG4_A.fa', '/tmp/1YT4_A.fa', '/tmp/1PZP_A.fa', '/tmp/4ZJ1_A.fa', '/tmp/8DDZ_A.fa', '/tmp/4RX3_A.fa', '/tmp/4RVA_A.fa', '/tmp/4OP5_A.fa', '/tmp/4ID4_A.fa']\n"
     ]
    }
   ],
   "source": [
    "from tempfile import gettempdir, NamedTemporaryFile\n",
    "import biotite.database.entrez as entrez\n",
    "import pandas as pd\n",
    "from fuzzywuzzy import fuzz\n",
    "# going back from e.g 4ID4 to 4ID4_A for each of the PDB ids, due to databases differences\n",
    "#in accession id requirement\n",
    "index = minPDB.reset_index()\n",
    "#index = index['pdb_code']\n",
    "#print(index)\n",
    "\n",
    "similar_values = []\n",
    "\n",
    "# Iterate over the rows in both dataframes\n",
    "for index, row in df.iterrows():\n",
    "    value1 = row[\"accession\"]\n",
    "    for index, row1 in minPDB.iterrows():\n",
    "        value2 = row1[\"pdb_code\"]\n",
    "        # Calculate the similarity between the two values\n",
    "        similarity = fuzz.token_set_ratio(value1, value2)\n",
    "        # If the similarity is greater than a certain threshold, add the values to the list\n",
    "        if similarity > 70:\n",
    "            similar_values.append((value1))\n",
    "\n",
    "# View the list of similar values\n",
    "#print(similar_values)\n",
    "file_paths = entrez.fetch(similar_values, gettempdir(), suffix=\"fa\",db_name=\"protein\", ret_type=\"fasta\"\n",
    ")\n",
    "print(file_paths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bb8db67f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import the required modules:\n",
    "import os\n",
    "import glob\n",
    "\n",
    "#Use the glob module to find all the fasta files in the tmp folder:\n",
    "fasta_files = glob.glob(\"/tmp/*.fa\")\n",
    "\n",
    "#Open the output file where you want to save the combined fasta file:\n",
    "with open(\"TEM-1hmm.faa\", \"w\") as outfile:\n",
    "    \n",
    "#Iterate over the list of fasta files and write their contents to the output file:\n",
    "    for fname in fasta_files:\n",
    "        with open(fname) as infile:\n",
    "            outfile.write(infile.read())\n",
    "\n",
    "#Close the output file:\n",
    "outfile.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5848ca6f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment with 10 rows and 297 columns\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|1YT4|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|1LI9|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|1PZP|A\n",
      "MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGY...HHH pdb|4ZJ1|A\n",
      "MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGY...--- pdb|1ZG4|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|4RVA|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|4RX3|A\n",
      "----------------------GHPETLVKVKDAEDQLGARVGY...--- pdb|8DDZ|A\n",
      "-----------------------HPETLVKVKDAEDQLGGRVGY...--- pdb|4OP5|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|4ID4|A\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio.Align.Applications import ClustalwCommandline\n",
    "from Bio import AlignIO, SeqIO\n",
    "\n",
    "# Run Clustalw to align the sequences in the FASTA file\n",
    "clustalw_cline = ClustalwCommandline(\"clustalw\", infile=\"TEM-1hmm.faa\")\n",
    "stdout, stderr = clustalw_cline()\n",
    "\n",
    "# Read the alignment from the Clustalw output file\n",
    "alignment = AlignIO.read(\"TEM-1hmm.aln\", \"clustal\")\n",
    "\n",
    "# Print the alignment\n",
    "print(alignment)\n",
    "AlignIO.write(alignment, \"alignment.sto\", \"stockholm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "71941d6b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment with 10 rows and 297 columns\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|1YT4|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|1LI9|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|1PZP|A\n",
      "MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGY...HHH pdb|4ZJ1|A\n",
      "MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGY...--- pdb|1ZG4|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|4RVA|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|4RX3|A\n",
      "----------------------GHPETLVKVKDAEDQLGARVGY...--- pdb|8DDZ|A\n",
      "-----------------------HPETLVKVKDAEDQLGGRVGY...--- pdb|4OP5|A\n",
      "-----------------------HPETLVKVKDAEDQLGARVGY...--- pdb|4ID4|A\n"
     ]
    }
   ],
   "source": [
    "import pyhmmer\n",
    "from Bio import SeqIO\n",
    "from Bio import AlignIO\n",
    "pyhmmer.__version__\n",
    "\n",
    "align = AlignIO.read(\"alignment.sto\", \"stockholm\")\n",
    "print(align)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "64df1bc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create amino acid alphabet\n",
    "alphabet = pyhmmer.easel.Alphabet.amino()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c676d71e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pyhmmer.easel.DigitalMSA object at 0x7f2fb926b200>\n"
     ]
    }
   ],
   "source": [
    "#read in the msa\n",
    "with pyhmmer.easel.MSAFile(\"alignment.sto\", digital=True, alphabet=alphabet) as msa_file:\n",
    "    msa = msa_file.read()\n",
    "    print(msa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ed130415",
   "metadata": {},
   "outputs": [],
   "source": [
    "#this entire function build the HMM\n",
    "#important note, inside the created \"alignment.sto\" file you need on the second line \n",
    "#to add a name for the pyhmmer package to create an HMM e.g : #=GF ID TEM-1\n",
    "#1.# STOCKHOLM 1.0\n",
    "#2.#=GF ID TEM-1 (this is the line)\n",
    "#3.#=GF SQ 10\n",
    "builder = pyhmmer.plan7.Builder(alphabet)\n",
    "background = pyhmmer.plan7.Background(alphabet)\n",
    "hmm, _, _ = builder.build_msa(msa, background)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "74e8fddc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'hpetlvkvkdaedqlgarvgyieldlnsgkilesfrpeerfpmlstfkvllcgavlsrvdagqeqlgrrihysqndlveyspvtekhltdgmtvrelcsaaitmsdntaanlllttiggpkeltaflhnlgdhvtrldryepelneaipnderdttmpaalattlrklltgelltlasrqqlidwmeadkvagpllrsalpagwfiadksgagergsrgiiaalgpdgkpsrivviyttgsqatmdernrqiaeigaslikhw'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#just to see the consensus sequence of the HMM\n",
    "hmm.consensus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4a3b51ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "#save the HMM file\n",
    "with open(\"TEM-1.hmm\", \"wb\") as output_file:\n",
    "    hmm.write(output_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6ecdb1b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pyhmmer\n",
    "#run and compare the query sequence to the HMM\n",
    "pipeline = pyhmmer.plan7.Pipeline(alphabet, background=background)\n",
    "with pyhmmer.easel.SequenceFile(\"sequence.842.fasta\", digital=True, alphabet=alphabet) as seq_file:\n",
    "    hits = pipeline.search_hmm(hmm, seq_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "560c72f1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                 TEM-1   1 hpetlvkvkdaedqlgarvgyieldlnsgkilesfrpeerfpmlstfkvllcgavlsrvdagqeqlgrrihysqndlveyspvtekhltdgmtvrelcsaaitmsdntaanlllttiggpkeltaflhnlgdhvtrldryepelneaipnderdttmpaalattlrklltgelltlasrqqlidwmeadkvagpllrsalpagwfiadksgagergsrgiiaalgpdgkpsrivviyttgsqatmdernrqiaeigaslikhw 263\n",
      "                                                           hpetlvkvkdaedqlgarvgyieldlnsgkilesfrpeerfpm+stfkvllcgavlsrvdagqeqlgrrihysqndlveyspvtekhltdgmtvrelcsaaitmsdntaanlllttiggpkeltaflhn+gdhvtrldr+epelneaipnderdttmpaa+attlrklltgelltlasrqqlidwmeadkvagpllrsalpagwfiadksgagergsrgiiaalgpdgkpsrivviyttgsqatmdernrqiaeigaslikhw\n",
      "  sid|842|pid|818|hfid|1|sfid|1|gi|46015634|taxid|562|   1 HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFKVLLCGAVLSRVDAGQEQLGRRIHYSQNDLVEYSPVTEKHLTDGMTVRELCSAAITMSDNTAANLLLTTIGGPKELTAFLHNMGDHVTRLDRWEPELNEAIPNDERDTTMPAAMATTLRKLLTGELLTLASRQQLIDWMEADKVAGPLLRSALPAGWFIADKSGAGERGSRGIIAALGPDGKPSRIVVIYTTGSQATMDERNRQIAEIGASLIKHW 263\n",
      "                                                           9********************************************************************************************************************************************************************************************************************************************************************** PP\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ali = hits[0].domains[0].alignment\n",
    "print(ali)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "d8c327de",
   "metadata": {},
   "outputs": [],
   "source": [
    "#save into a summarized table\n",
    "with open(\"TEM-1.domtbl\", \"wb\") as f:\n",
    "    hits.write(f, format=\"domains\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "91b05e14",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------------HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFKVLLCGAVLSRVDAGQEQLGRRIHYSQNDLVEYSPVTEKHLTDGMTVRELCSAAITMSDNTAANLLLTTIGGPKELTAFLHNMGDHVTRLDRWEPELNEAIPNDERDTTMPAAMATTLRKLLTGELLTLASRQQLIDWMEADKVAGPLLRSALPAGWFIADKSGAGERGSRGIIAALGPDGKPSRIVVIYTTGSQATMDERNRQIAEIGASLIKHW-----------\n",
      "                       ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|||||||||||||||||||||||||||||||.||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||           \n",
      "MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFKVLLCGAVLSRIDAGQEQLGRRIHYSQNDLVEYSPVTEKHLTDGMTVRELCSAAITMSDNTAANLLLTTIGGPKELTAFLHNMGDHVTRLDRWEPELNEAIPNDERDTTMPVAMATTLRKLLTGELLTLASRQQLIDWMEADKXAGPLLRSALPAGWFIADKSGAGERGSRGIIAALGPDGKPSRIVVIYTTGSQATMDERNRQIAEIGASLIKHWAAALEHHHHHH\n",
      "  Score=518.8\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment\n",
    "# Read the sequences from the fasta files\n",
    "with open(\"sequence.842.fasta\", \"r\") as handle:\n",
    "    seq1 = SeqIO.read(handle, \"fasta\").seq\n",
    "with open(\"4ZJ1_A.fa\", \"r\") as handle:\n",
    "    seq2 = SeqIO.read(handle, \"fasta\").seq\n",
    "\n",
    "# Perform the alignment\n",
    "for a in pairwise2.align.globalms(seq1, seq2, 2, 1, -0.5, -0.1):\n",
    "    print(format_alignment(*a))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "266f654f",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment\n",
    "import io\n",
    "import os\n",
    "# Read the sequences from the fasta files\n",
    "with open(\"sequence.842.fasta\", \"r\") as handle:\n",
    "    seq1 = SeqIO.read(handle, \"fasta\").seq\n",
    "with open(\"4ZJ1_A.fa\", \"r\") as handle:\n",
    "    seq2 = SeqIO.read(handle, \"fasta\").seq\n",
    "\n",
    "# Perform the alignment\n",
    "alignments = pairwise2.align.globalms(seq1, seq2, 2, 1, -0.5, -0.1)\n",
    "\n",
    "best_alignment = alignments[0]\n",
    "\n",
    "# extract the aligned sequences\n",
    "aligned_seq1 = best_alignment[0]\n",
    "aligned_seq2 = best_alignment[1]\n",
    "\n",
    "# save the 2 alignments\n",
    "#with open(\"aligned_seq1.txt\", \"a\") as f:\n",
    "   # print(aligned_seq1, file=f)\n",
    "#with open(\"aligned_seq2.txt\", \"a\") as f:\n",
    "    #print(aligned_seq2, file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "c87a3bb9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " amino number standard numbering\n",
      "            1                  -\n",
      "            2                  -\n",
      "            3                  -\n",
      "            4                  -\n",
      "            5                  -\n",
      "            6                  -\n",
      "            7                  -\n",
      "            8                  -\n",
      "            9                  -\n",
      "           10                  -\n",
      "           11                  -\n",
      "           12                  -\n",
      "           13                  -\n",
      "           14                  -\n",
      "           15                  -\n",
      "           16                  -\n",
      "           17                  -\n",
      "           18                  -\n",
      "           19                  -\n",
      "           20                  -\n",
      "           21                  -\n",
      "           22                  -\n",
      "           23                  -\n",
      "           24                  H\n",
      "           25                  P\n",
      "           26                  E\n",
      "           27                  T\n",
      "           28                  L\n",
      "           29                  V\n",
      "           30                  K\n",
      "           31                  V\n",
      "           32                  K\n",
      "           33                  D\n",
      "           34                  A\n",
      "           35                  E\n",
      "           36                  D\n",
      "           37                  Q\n",
      "           38                  L\n",
      "           39                  G\n",
      "           40                  A\n",
      "           41                  R\n",
      "           42                  V\n",
      "           43                  G\n",
      "           44                  Y\n",
      "           45                  I\n",
      "           46                  E\n",
      "           47                  L\n",
      "           48                  D\n",
      "           49                  L\n",
      "           50                  N\n",
      "           51                  S\n",
      "           52                  G\n",
      "           53                  K\n",
      "           54                  I\n",
      "           55                  L\n",
      "           56                  E\n",
      "           57                  S\n",
      "           58                  F\n",
      "           59                  R\n",
      "           60                  P\n",
      "           61                  E\n",
      "           62                  E\n",
      "           63                  R\n",
      "           64                  F\n",
      "           65                  P\n",
      "           66                  M\n",
      "           67                  M\n",
      "           68                  S\n",
      "           69                  T\n",
      "           70                  F\n",
      "           71                  K\n",
      "           72                  V\n",
      "           73                  L\n",
      "           74                  L\n",
      "           75                  C\n",
      "           76                  G\n",
      "           77                  A\n",
      "           78                  V\n",
      "           79                  L\n",
      "           80                  S\n",
      "           81                  R\n",
      "           82                  V\n",
      "           83                  D\n",
      "           84                  A\n",
      "           85                  G\n",
      "           86                  Q\n",
      "           87                  E\n",
      "           88                  Q\n",
      "           89                  L\n",
      "           90                  G\n",
      "           91                  R\n",
      "           92                  R\n",
      "           93                  I\n",
      "           94                  H\n",
      "           95                  Y\n",
      "           96                  S\n",
      "           97                  Q\n",
      "           98                  N\n",
      "           99                  D\n",
      "          100                  L\n",
      "          101                  V\n",
      "          102                  E\n",
      "          103                  Y\n",
      "          104                  S\n",
      "          105                  P\n",
      "          106                  V\n",
      "          107                  T\n",
      "          108                  E\n",
      "          109                  K\n",
      "          110                  H\n",
      "          111                  L\n",
      "          112                  T\n",
      "          113                  D\n",
      "          114                  G\n",
      "          115                  M\n",
      "          116                  T\n",
      "          117                  V\n",
      "          118                  R\n",
      "          119                  E\n",
      "          120                  L\n",
      "          121                  C\n",
      "          122                  S\n",
      "          123                  A\n",
      "          124                  A\n",
      "          125                  I\n",
      "          126                  T\n",
      "          127                  M\n",
      "          128                  S\n",
      "          129                  D\n",
      "          130                  N\n",
      "          131                  T\n",
      "          132                  A\n",
      "          133                  A\n",
      "          134                  N\n",
      "          135                  L\n",
      "          136                  L\n",
      "          137                  L\n",
      "          138                  T\n",
      "          139                  T\n",
      "          140                  I\n",
      "          141                  G\n",
      "          142                  G\n",
      "          143                  P\n",
      "          144                  K\n",
      "          145                  E\n",
      "          146                  L\n",
      "          147                  T\n",
      "          148                  A\n",
      "          149                  F\n",
      "          150                  L\n",
      "          151                  H\n",
      "          152                  N\n",
      "          153                  M\n",
      "          154                  G\n",
      "          155                  D\n",
      "          156                  H\n",
      "          157                  V\n",
      "          158                  T\n",
      "          159                  R\n",
      "          160                  L\n",
      "          161                  D\n",
      "          162                  R\n",
      "          163                  W\n",
      "          164                  E\n",
      "          165                  P\n",
      "          166                  E\n",
      "          167                  L\n",
      "          168                  N\n",
      "          169                  E\n",
      "          170                  A\n",
      "          171                  I\n",
      "          172                  P\n",
      "          173                  N\n",
      "          174                  D\n",
      "          175                  E\n",
      "          176                  R\n",
      "          177                  D\n",
      "          178                  T\n",
      "          179                  T\n",
      "          180                  M\n",
      "          181                  P\n",
      "          182                  A\n",
      "          183                  A\n",
      "          184                  M\n",
      "          185                  A\n",
      "          186                  T\n",
      "          187                  T\n",
      "          188                  L\n",
      "          189                  R\n",
      "          190                  K\n",
      "          191                  L\n",
      "          192                  L\n",
      "          193                  T\n",
      "          194                  G\n",
      "          195                  E\n",
      "          196                  L\n",
      "          197                  L\n",
      "          198                  T\n",
      "          199                  L\n",
      "          200                  A\n",
      "          201                  S\n",
      "          202                  R\n",
      "          203                  Q\n",
      "          204                  Q\n",
      "          205                  L\n",
      "          206                  I\n",
      "          207                  D\n",
      "          208                  W\n",
      "          209                  M\n",
      "          210                  E\n",
      "          211                  A\n",
      "          212                  D\n",
      "          213                  K\n",
      "          214                  V\n",
      "          215                  A\n",
      "          216                  G\n",
      "          217                  P\n",
      "          218                  L\n",
      "          219                  L\n",
      "          220                  R\n",
      "          221                  S\n",
      "          222                  A\n",
      "          223                  L\n",
      "          224                  P\n",
      "          225                  A\n",
      "          226                  G\n",
      "          227                  W\n",
      "          228                  F\n",
      "          229                  I\n",
      "          230                  A\n",
      "          231                  D\n",
      "          232                  K\n",
      "          233                  S\n",
      "          234                  G\n",
      "          235                  A\n",
      "          236                  G\n",
      "          237                  E\n",
      "          238                  R\n",
      "          239                  G\n",
      "          240                  S\n",
      "          241                  R\n",
      "          242                  G\n",
      "          243                  I\n",
      "          244                  I\n",
      "          245                  A\n",
      "          246                  A\n",
      "          247                  L\n",
      "          248                  G\n",
      "          249                  P\n",
      "          250                  D\n",
      "          251                  G\n",
      "          252                  K\n",
      "          253                  P\n",
      "          254                  S\n",
      "          255                  R\n",
      "          256                  I\n",
      "          257                  V\n",
      "          258                  V\n",
      "          259                  I\n",
      "          260                  Y\n",
      "          261                  T\n",
      "          262                  T\n",
      "          263                  G\n",
      "          264                  S\n",
      "          265                  Q\n",
      "          266                  A\n",
      "          267                  T\n",
      "          268                  M\n",
      "          269                  D\n",
      "          270                  E\n",
      "          271                  R\n",
      "          272                  N\n",
      "          273                  R\n",
      "          274                  Q\n",
      "          275                  I\n",
      "          276                  A\n",
      "          277                  E\n",
      "          278                  I\n",
      "          279                  G\n",
      "          280                  A\n",
      "          281                  S\n",
      "          282                  L\n",
      "          283                  I\n",
      "          284                  K\n",
      "          285                  H\n",
      "          286                  W\n",
      "          287                  -\n",
      "          288                  -\n",
      "          289                  -\n",
      "          290                  -\n",
      "          291                  -\n",
      "          292                  -\n",
      "          293                  -\n",
      "          294                  -\n",
      "          295                  -\n",
      "          296                  -\n",
      "          297                  -\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Open the FASTA file and read it into a string 4ZJ1_A.fa\n",
    "lines = aligned_seq1.split(\"\\n\")\n",
    "# Split the remaining lines into a list of individual amino acids\n",
    "amino_acids = [aa for line in lines for aa in line]\n",
    "# Enumerate the list of amino acids, starting from 1\n",
    "amino_acids_with_index = [(i+1, aa) for i, aa in enumerate(amino_acids)]\n",
    "# Print the list of amino acids with their indices\n",
    "#print(amino_acids_with_index)\n",
    "numberingdf = pd.DataFrame(amino_acids_with_index, columns=[\"amino number\",\"standard numbering\"])\n",
    "# Print the dataframe\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.width', None)\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "print(numberingdf.to_string(index=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "b960c83c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4ZJ1_A\n",
      "     M\n",
      "     S\n",
      "     I\n",
      "     Q\n",
      "     H\n",
      "     F\n",
      "     R\n",
      "     V\n",
      "     A\n",
      "     L\n",
      "     I\n",
      "     P\n",
      "     F\n",
      "     F\n",
      "     A\n",
      "     A\n",
      "     F\n",
      "     C\n",
      "     L\n",
      "     P\n",
      "     V\n",
      "     F\n",
      "     A\n",
      "     H\n",
      "     P\n",
      "     E\n",
      "     T\n",
      "     L\n",
      "     V\n",
      "     K\n",
      "     V\n",
      "     K\n",
      "     D\n",
      "     A\n",
      "     E\n",
      "     D\n",
      "     Q\n",
      "     L\n",
      "     G\n",
      "     A\n",
      "     R\n",
      "     V\n",
      "     G\n",
      "     Y\n",
      "     I\n",
      "     E\n",
      "     L\n",
      "     D\n",
      "     L\n",
      "     N\n",
      "     S\n",
      "     G\n",
      "     K\n",
      "     I\n",
      "     L\n",
      "     E\n",
      "     S\n",
      "     F\n",
      "     R\n",
      "     P\n",
      "     E\n",
      "     E\n",
      "     R\n",
      "     F\n",
      "     P\n",
      "     M\n",
      "     M\n",
      "     S\n",
      "     T\n",
      "     F\n",
      "     K\n",
      "     V\n",
      "     L\n",
      "     L\n",
      "     C\n",
      "     G\n",
      "     A\n",
      "     V\n",
      "     L\n",
      "     S\n",
      "     R\n",
      "     I\n",
      "     D\n",
      "     A\n",
      "     G\n",
      "     Q\n",
      "     E\n",
      "     Q\n",
      "     L\n",
      "     G\n",
      "     R\n",
      "     R\n",
      "     I\n",
      "     H\n",
      "     Y\n",
      "     S\n",
      "     Q\n",
      "     N\n",
      "     D\n",
      "     L\n",
      "     V\n",
      "     E\n",
      "     Y\n",
      "     S\n",
      "     P\n",
      "     V\n",
      "     T\n",
      "     E\n",
      "     K\n",
      "     H\n",
      "     L\n",
      "     T\n",
      "     D\n",
      "     G\n",
      "     M\n",
      "     T\n",
      "     V\n",
      "     R\n",
      "     E\n",
      "     L\n",
      "     C\n",
      "     S\n",
      "     A\n",
      "     A\n",
      "     I\n",
      "     T\n",
      "     M\n",
      "     S\n",
      "     D\n",
      "     N\n",
      "     T\n",
      "     A\n",
      "     A\n",
      "     N\n",
      "     L\n",
      "     L\n",
      "     L\n",
      "     T\n",
      "     T\n",
      "     I\n",
      "     G\n",
      "     G\n",
      "     P\n",
      "     K\n",
      "     E\n",
      "     L\n",
      "     T\n",
      "     A\n",
      "     F\n",
      "     L\n",
      "     H\n",
      "     N\n",
      "     M\n",
      "     G\n",
      "     D\n",
      "     H\n",
      "     V\n",
      "     T\n",
      "     R\n",
      "     L\n",
      "     D\n",
      "     R\n",
      "     W\n",
      "     E\n",
      "     P\n",
      "     E\n",
      "     L\n",
      "     N\n",
      "     E\n",
      "     A\n",
      "     I\n",
      "     P\n",
      "     N\n",
      "     D\n",
      "     E\n",
      "     R\n",
      "     D\n",
      "     T\n",
      "     T\n",
      "     M\n",
      "     P\n",
      "     V\n",
      "     A\n",
      "     M\n",
      "     A\n",
      "     T\n",
      "     T\n",
      "     L\n",
      "     R\n",
      "     K\n",
      "     L\n",
      "     L\n",
      "     T\n",
      "     G\n",
      "     E\n",
      "     L\n",
      "     L\n",
      "     T\n",
      "     L\n",
      "     A\n",
      "     S\n",
      "     R\n",
      "     Q\n",
      "     Q\n",
      "     L\n",
      "     I\n",
      "     D\n",
      "     W\n",
      "     M\n",
      "     E\n",
      "     A\n",
      "     D\n",
      "     K\n",
      "     X\n",
      "     A\n",
      "     G\n",
      "     P\n",
      "     L\n",
      "     L\n",
      "     R\n",
      "     S\n",
      "     A\n",
      "     L\n",
      "     P\n",
      "     A\n",
      "     G\n",
      "     W\n",
      "     F\n",
      "     I\n",
      "     A\n",
      "     D\n",
      "     K\n",
      "     S\n",
      "     G\n",
      "     A\n",
      "     G\n",
      "     E\n",
      "     R\n",
      "     G\n",
      "     S\n",
      "     R\n",
      "     G\n",
      "     I\n",
      "     I\n",
      "     A\n",
      "     A\n",
      "     L\n",
      "     G\n",
      "     P\n",
      "     D\n",
      "     G\n",
      "     K\n",
      "     P\n",
      "     S\n",
      "     R\n",
      "     I\n",
      "     V\n",
      "     V\n",
      "     I\n",
      "     Y\n",
      "     T\n",
      "     T\n",
      "     G\n",
      "     S\n",
      "     Q\n",
      "     A\n",
      "     T\n",
      "     M\n",
      "     D\n",
      "     E\n",
      "     R\n",
      "     N\n",
      "     R\n",
      "     Q\n",
      "     I\n",
      "     A\n",
      "     E\n",
      "     I\n",
      "     G\n",
      "     A\n",
      "     S\n",
      "     L\n",
      "     I\n",
      "     K\n",
      "     H\n",
      "     W\n",
      "     A\n",
      "     A\n",
      "     A\n",
      "     L\n",
      "     E\n",
      "     H\n",
      "     H\n",
      "     H\n",
      "     H\n",
      "     H\n",
      "     H\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Open the FASTA file and read it into a string 4ZJ1_A.fa\n",
    "lines = aligned_seq2.split(\"\\n\")\n",
    "# Split the remaining lines into a list of individual amino acids\n",
    "amino_acids = [aa for line in lines for aa in line]\n",
    "# Enumerate the list of amino acids, starting from 1\n",
    "amino_acids_with_index = [(i+1, aa) for i, aa in enumerate(amino_acids)]\n",
    "# Print the list of amino acids with their indices\n",
    "#print(amino_acids_with_index)\n",
    "numberingdf2 = pd.DataFrame(amino_acids, columns=[\"4ZJ1_A\"])\n",
    "# Print the dataframe\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)\n",
    "pd.set_option('display.width', None)\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "print(numberingdf2.to_string(index=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "1c07b758",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " amino number standard numbering 4ZJ1_A new_index\n",
      "            1                  -      M       0.1\n",
      "            2                  -      S       0.2\n",
      "            3                  -      I       0.3\n",
      "            4                  -      Q       0.4\n",
      "            5                  -      H       0.5\n",
      "            6                  -      F       0.6\n",
      "            7                  -      R       0.7\n",
      "            8                  -      V       0.8\n",
      "            9                  -      A       0.9\n",
      "           10                  -      L      0.10\n",
      "           11                  -      I      0.11\n",
      "           12                  -      P      0.12\n",
      "           13                  -      F      0.13\n",
      "           14                  -      F      0.14\n",
      "           15                  -      A      0.15\n",
      "           16                  -      A      0.16\n",
      "           17                  -      F      0.17\n",
      "           18                  -      C      0.18\n",
      "           19                  -      L      0.19\n",
      "           20                  -      P      0.20\n",
      "           21                  -      V      0.21\n",
      "           22                  -      F      0.22\n",
      "           23                  -      A      0.23\n",
      "           24                  H      H         1\n",
      "           25                  P      P         2\n",
      "           26                  E      E         3\n",
      "           27                  T      T         4\n",
      "           28                  L      L         5\n",
      "           29                  V      V         6\n",
      "           30                  K      K         7\n",
      "           31                  V      V         8\n",
      "           32                  K      K         9\n",
      "           33                  D      D        10\n",
      "           34                  A      A        11\n",
      "           35                  E      E        12\n",
      "           36                  D      D        13\n",
      "           37                  Q      Q        14\n",
      "           38                  L      L        15\n",
      "           39                  G      G        16\n",
      "           40                  A      A        17\n",
      "           41                  R      R        18\n",
      "           42                  V      V        19\n",
      "           43                  G      G        20\n",
      "           44                  Y      Y        21\n",
      "           45                  I      I        22\n",
      "           46                  E      E        23\n",
      "           47                  L      L        24\n",
      "           48                  D      D        25\n",
      "           49                  L      L        26\n",
      "           50                  N      N        27\n",
      "           51                  S      S        28\n",
      "           52                  G      G        29\n",
      "           53                  K      K        30\n",
      "           54                  I      I        31\n",
      "           55                  L      L        32\n",
      "           56                  E      E        33\n",
      "           57                  S      S        34\n",
      "           58                  F      F        35\n",
      "           59                  R      R        36\n",
      "           60                  P      P        37\n",
      "           61                  E      E        38\n",
      "           62                  E      E        39\n",
      "           63                  R      R        40\n",
      "           64                  F      F        41\n",
      "           65                  P      P        42\n",
      "           66                  M      M        43\n",
      "           67                  M      M        44\n",
      "           68                  S      S        45\n",
      "           69                  T      T        46\n",
      "           70                  F      F        47\n",
      "           71                  K      K        48\n",
      "           72                  V      V        49\n",
      "           73                  L      L        50\n",
      "           74                  L      L        51\n",
      "           75                  C      C        52\n",
      "           76                  G      G        53\n",
      "           77                  A      A        54\n",
      "           78                  V      V        55\n",
      "           79                  L      L        56\n",
      "           80                  S      S        57\n",
      "           81                  R      R        58\n",
      "           82                  V      I        59\n",
      "           83                  D      D        60\n",
      "           84                  A      A        61\n",
      "           85                  G      G        62\n",
      "           86                  Q      Q        63\n",
      "           87                  E      E        64\n",
      "           88                  Q      Q        65\n",
      "           89                  L      L        66\n",
      "           90                  G      G        67\n",
      "           91                  R      R        68\n",
      "           92                  R      R        69\n",
      "           93                  I      I        70\n",
      "           94                  H      H        71\n",
      "           95                  Y      Y        72\n",
      "           96                  S      S        73\n",
      "           97                  Q      Q        74\n",
      "           98                  N      N        75\n",
      "           99                  D      D        76\n",
      "          100                  L      L        77\n",
      "          101                  V      V        78\n",
      "          102                  E      E        79\n",
      "          103                  Y      Y        80\n",
      "          104                  S      S        81\n",
      "          105                  P      P        82\n",
      "          106                  V      V        83\n",
      "          107                  T      T        84\n",
      "          108                  E      E        85\n",
      "          109                  K      K        86\n",
      "          110                  H      H        87\n",
      "          111                  L      L        88\n",
      "          112                  T      T        89\n",
      "          113                  D      D        90\n",
      "          114                  G      G        91\n",
      "          115                  M      M        92\n",
      "          116                  T      T        93\n",
      "          117                  V      V        94\n",
      "          118                  R      R        95\n",
      "          119                  E      E        96\n",
      "          120                  L      L        97\n",
      "          121                  C      C        98\n",
      "          122                  S      S        99\n",
      "          123                  A      A       100\n",
      "          124                  A      A       101\n",
      "          125                  I      I       102\n",
      "          126                  T      T       103\n",
      "          127                  M      M       104\n",
      "          128                  S      S       105\n",
      "          129                  D      D       106\n",
      "          130                  N      N       107\n",
      "          131                  T      T       108\n",
      "          132                  A      A       109\n",
      "          133                  A      A       110\n",
      "          134                  N      N       111\n",
      "          135                  L      L       112\n",
      "          136                  L      L       113\n",
      "          137                  L      L       114\n",
      "          138                  T      T       115\n",
      "          139                  T      T       116\n",
      "          140                  I      I       117\n",
      "          141                  G      G       118\n",
      "          142                  G      G       119\n",
      "          143                  P      P       120\n",
      "          144                  K      K       121\n",
      "          145                  E      E       122\n",
      "          146                  L      L       123\n",
      "          147                  T      T       124\n",
      "          148                  A      A       125\n",
      "          149                  F      F       126\n",
      "          150                  L      L       127\n",
      "          151                  H      H       128\n",
      "          152                  N      N       129\n",
      "          153                  M      M       130\n",
      "          154                  G      G       131\n",
      "          155                  D      D       132\n",
      "          156                  H      H       133\n",
      "          157                  V      V       134\n",
      "          158                  T      T       135\n",
      "          159                  R      R       136\n",
      "          160                  L      L       137\n",
      "          161                  D      D       138\n",
      "          162                  R      R       139\n",
      "          163                  W      W       140\n",
      "          164                  E      E       141\n",
      "          165                  P      P       142\n",
      "          166                  E      E       143\n",
      "          167                  L      L       144\n",
      "          168                  N      N       145\n",
      "          169                  E      E       146\n",
      "          170                  A      A       147\n",
      "          171                  I      I       148\n",
      "          172                  P      P       149\n",
      "          173                  N      N       150\n",
      "          174                  D      D       151\n",
      "          175                  E      E       152\n",
      "          176                  R      R       153\n",
      "          177                  D      D       154\n",
      "          178                  T      T       155\n",
      "          179                  T      T       156\n",
      "          180                  M      M       157\n",
      "          181                  P      P       158\n",
      "          182                  A      V       159\n",
      "          183                  A      A       160\n",
      "          184                  M      M       161\n",
      "          185                  A      A       162\n",
      "          186                  T      T       163\n",
      "          187                  T      T       164\n",
      "          188                  L      L       165\n",
      "          189                  R      R       166\n",
      "          190                  K      K       167\n",
      "          191                  L      L       168\n",
      "          192                  L      L       169\n",
      "          193                  T      T       170\n",
      "          194                  G      G       171\n",
      "          195                  E      E       172\n",
      "          196                  L      L       173\n",
      "          197                  L      L       174\n",
      "          198                  T      T       175\n",
      "          199                  L      L       176\n",
      "          200                  A      A       177\n",
      "          201                  S      S       178\n",
      "          202                  R      R       179\n",
      "          203                  Q      Q       180\n",
      "          204                  Q      Q       181\n",
      "          205                  L      L       182\n",
      "          206                  I      I       183\n",
      "          207                  D      D       184\n",
      "          208                  W      W       185\n",
      "          209                  M      M       186\n",
      "          210                  E      E       187\n",
      "          211                  A      A       188\n",
      "          212                  D      D       189\n",
      "          213                  K      K       190\n",
      "          214                  V      X       191\n",
      "          215                  A      A       192\n",
      "          216                  G      G       193\n",
      "          217                  P      P       194\n",
      "          218                  L      L       195\n",
      "          219                  L      L       196\n",
      "          220                  R      R       197\n",
      "          221                  S      S       198\n",
      "          222                  A      A       199\n",
      "          223                  L      L       200\n",
      "          224                  P      P       201\n",
      "          225                  A      A       202\n",
      "          226                  G      G       203\n",
      "          227                  W      W       204\n",
      "          228                  F      F       205\n",
      "          229                  I      I       206\n",
      "          230                  A      A       207\n",
      "          231                  D      D       208\n",
      "          232                  K      K       209\n",
      "          233                  S      S       210\n",
      "          234                  G      G       211\n",
      "          235                  A      A       212\n",
      "          236                  G      G       213\n",
      "          237                  E      E       214\n",
      "          238                  R      R       215\n",
      "          239                  G      G       216\n",
      "          240                  S      S       217\n",
      "          241                  R      R       218\n",
      "          242                  G      G       219\n",
      "          243                  I      I       220\n",
      "          244                  I      I       221\n",
      "          245                  A      A       222\n",
      "          246                  A      A       223\n",
      "          247                  L      L       224\n",
      "          248                  G      G       225\n",
      "          249                  P      P       226\n",
      "          250                  D      D       227\n",
      "          251                  G      G       228\n",
      "          252                  K      K       229\n",
      "          253                  P      P       230\n",
      "          254                  S      S       231\n",
      "          255                  R      R       232\n",
      "          256                  I      I       233\n",
      "          257                  V      V       234\n",
      "          258                  V      V       235\n",
      "          259                  I      I       236\n",
      "          260                  Y      Y       237\n",
      "          261                  T      T       238\n",
      "          262                  T      T       239\n",
      "          263                  G      G       240\n",
      "          264                  S      S       241\n",
      "          265                  Q      Q       242\n",
      "          266                  A      A       243\n",
      "          267                  T      T       244\n",
      "          268                  M      M       245\n",
      "          269                  D      D       246\n",
      "          270                  E      E       247\n",
      "          271                  R      R       248\n",
      "          272                  N      N       249\n",
      "          273                  R      R       250\n",
      "          274                  Q      Q       251\n",
      "          275                  I      I       252\n",
      "          276                  A      A       253\n",
      "          277                  E      E       254\n",
      "          278                  I      I       255\n",
      "          279                  G      G       256\n",
      "          280                  A      A       257\n",
      "          281                  S      S       258\n",
      "          282                  L      L       259\n",
      "          283                  I      I       260\n",
      "          284                  K      K       261\n",
      "          285                  H      H       262\n",
      "          286                  W      W       263\n",
      "          287                  -      A     263.1\n",
      "          288                  -      A     263.2\n",
      "          289                  -      A     263.3\n",
      "          290                  -      L     263.4\n",
      "          291                  -      E     263.5\n",
      "          292                  -      H     263.6\n",
      "          293                  -      H     263.7\n",
      "          294                  -      H     263.8\n",
      "          295                  -      H     263.9\n",
      "          296                  -      H    263.10\n",
      "          297                  -      H    263.11\n"
     ]
    }
   ],
   "source": [
    "combined_df = numberingdf.join(numberingdf2)\n",
    "counter = 0\n",
    "counternew = 0\n",
    "\n",
    "def renumber_index(row):\n",
    "    global counter\n",
    "    global counternew\n",
    "    new_index = 0\n",
    "    if row['standard numbering'] == '-':\n",
    "        counternew += 1\n",
    "        new_index = f\"{counter}.{counternew}\"\n",
    "    elif row['standard numbering'] != '-':\n",
    "        counter += 1 \n",
    "        counternew = 0\n",
    "        new_index = counter\n",
    "    return new_index\n",
    "\n",
    "combined_df['new_index'] = combined_df.apply(renumber_index, axis=1)\n",
    "\n",
    "print(combined_df.to_string(index=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "86c19f38",
   "metadata": {},
   "outputs": [],
   "source": [
    "from unipressed import UniprotkbClient\n",
    "import xml.etree.ElementTree as ET\n",
    "from dict2xml import dict2xml\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "import shutil\n",
    "import flatten_json\n",
    "#PDB to uniprot database \n",
    "#7LIV Structure of human transfer RNA visualized in the cytomegalovirus, a DNA virus\n",
    "for record in UniprotkbClient.search(\n",
    "    query={\"xref\": \"pdb-1S0W\"},\n",
    ").each_record():\n",
    "    #display(record)\n",
    "\n",
    "    flat_data = flatten_json.flatten(record)\n",
    "# features_0_location_end_value, features_7_type, features_0_description\n",
    "# Iterate over the items in the flattened dictionary\n",
    "    #for key, value in flat_data.items():\n",
    "        #print(key, value)\n",
    "    i = 1\n",
    "    while True:\n",
    "        key_start = (f\"features_{i}_location_start_value\")\n",
    "        key_end = (f\"features_{i}_location_end_value\")\n",
    "        key_type = (f\"features_{i}_type\")\n",
    "        key_feature = (f\"features_{i}_description\")\n",
    "\n",
    "        start = flat_data.get(key_start)\n",
    "        end = flat_data.get(key_end)\n",
    "        type_0 = flat_data.get(key_type)\n",
    "        feature = flat_data.get(key_feature)\n",
    "        if start is None:\n",
    "            break\n",
    "        if end is None:\n",
    "            break\n",
    "        if type_0 is None:\n",
    "            break\n",
    "        if feature is None:\n",
    "            break\n",
    "        #print(start ,end)\n",
    "        i += 1\n",
    "  \n",
    "        with open(\"annotationES.txt\", \"a\") as f:\n",
    "            print(start ,end, file=f)\n",
    "        with open(\"annotationT.txt\", \"a\") as f:\n",
    "            print(type_0, file=f)\n",
    "        with open(\"annotationF.txt\", \"a\") as f:\n",
    "            print(feature, file=f)\n",
    "            #P62593"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "3dab6d11",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     start  end             type  \\\n",
      "0       24  286            Chain   \n",
      "1       68   68      Active site   \n",
      "2      166  166      Active site   \n",
      "3      232  234     Binding site   \n",
      "4       75  121   Disulfide bond   \n",
      "5       19   19  Natural variant   \n",
      "6       37   37  Natural variant   \n",
      "7       67   67  Natural variant   \n",
      "8      102  102  Natural variant   \n",
      "9      162  162  Natural variant   \n",
      "10     162  162  Natural variant   \n",
      "11     235  235  Natural variant   \n",
      "12     236  236  Natural variant   \n",
      "13     237  237  Natural variant   \n",
      "14     261  261  Natural variant   \n",
      "15     272  272  Natural variant   \n",
      "16      25   38            Helix   \n",
      "17      40   48      Beta strand   \n",
      "18      49   51             Turn   \n",
      "19      54   59      Beta strand   \n",
      "20      67   69            Helix   \n",
      "21      70   83            Helix   \n",
      "22      97   99            Helix   \n",
      "23     107  109            Helix   \n",
      "24     111  113             Turn   \n",
      "25     117  126            Helix   \n",
      "26     130  139            Helix   \n",
      "27     143  152            Helix   \n",
      "28     166  168            Helix   \n",
      "29     171  173             Turn   \n",
      "30     181  193            Helix   \n",
      "31     194  197      Beta strand   \n",
      "32     199  210            Helix   \n",
      "33     213  215      Beta strand   \n",
      "34     216  218             Turn   \n",
      "35     219  222            Helix   \n",
      "36     228  235      Beta strand   \n",
      "37     237  239             Turn   \n",
      "38     241  249      Beta strand   \n",
      "39     255  263      Beta strand   \n",
      "40     264  266      Beta strand   \n",
      "41     268  284            Helix   \n",
      "42      37  201            Chain   \n",
      "43      37  112           Repeat   \n",
      "44     116  201           Repeat   \n",
      "45      66   78   Disulfide bond   \n",
      "46     145  167   Disulfide bond   \n",
      "47      42   47            Helix   \n",
      "48      54   61            Helix   \n",
      "49      63   65            Helix   \n",
      "50      66   69      Beta strand   \n",
      "51      70   72            Helix   \n",
      "52      75   83      Beta strand   \n",
      "53      86   98      Beta strand   \n",
      "54     103  111      Beta strand   \n",
      "55     121  126            Helix   \n",
      "56     133  140            Helix   \n",
      "57     143  151      Beta strand   \n",
      "58     153  156             Turn   \n",
      "59     162  168      Beta strand   \n",
      "60     170  172             Turn   \n",
      "61     175  177      Beta strand   \n",
      "62     180  188      Beta strand   \n",
      "63     191  199      Beta strand   \n",
      "64      24  286            Chain   \n",
      "65      68   68      Active site   \n",
      "66     166  166      Active site   \n",
      "67     232  234     Binding site   \n",
      "68      75  121   Disulfide bond   \n",
      "69      19   19  Natural variant   \n",
      "70      37   37  Natural variant   \n",
      "71      67   67  Natural variant   \n",
      "72     102  102  Natural variant   \n",
      "73     162  162  Natural variant   \n",
      "74     162  162  Natural variant   \n",
      "75     235  235  Natural variant   \n",
      "76     236  236  Natural variant   \n",
      "77     237  237  Natural variant   \n",
      "78     261  261  Natural variant   \n",
      "79     272  272  Natural variant   \n",
      "80      25   38            Helix   \n",
      "81      40   48      Beta strand   \n",
      "82      49   51             Turn   \n",
      "83      54   59      Beta strand   \n",
      "84      67   69            Helix   \n",
      "85      70   83            Helix   \n",
      "86      97   99            Helix   \n",
      "87     107  109            Helix   \n",
      "88     111  113             Turn   \n",
      "89     117  126            Helix   \n",
      "90     130  139            Helix   \n",
      "91     143  152            Helix   \n",
      "92     166  168            Helix   \n",
      "93     171  173             Turn   \n",
      "94     181  193            Helix   \n",
      "95     194  197      Beta strand   \n",
      "96     199  210            Helix   \n",
      "97     213  215      Beta strand   \n",
      "98     216  218             Turn   \n",
      "99     219  222            Helix   \n",
      "100    228  235      Beta strand   \n",
      "101    237  239             Turn   \n",
      "102    241  249      Beta strand   \n",
      "103    255  263      Beta strand   \n",
      "104    264  266      Beta strand   \n",
      "105    268  284            Helix   \n",
      "106     37  201            Chain   \n",
      "107     37  112           Repeat   \n",
      "108    116  201           Repeat   \n",
      "109     66   78   Disulfide bond   \n",
      "110    145  167   Disulfide bond   \n",
      "111     42   47            Helix   \n",
      "112     54   61            Helix   \n",
      "113     63   65            Helix   \n",
      "114     66   69      Beta strand   \n",
      "115     70   72            Helix   \n",
      "116     75   83      Beta strand   \n",
      "117     86   98      Beta strand   \n",
      "118    103  111      Beta strand   \n",
      "119    121  126            Helix   \n",
      "120    133  140            Helix   \n",
      "121    143  151      Beta strand   \n",
      "122    153  156             Turn   \n",
      "123    162  168      Beta strand   \n",
      "124    170  172             Turn   \n",
      "125    175  177      Beta strand   \n",
      "126    180  188      Beta strand   \n",
      "127    191  199      Beta strand   \n",
      "128     24  286            Chain   \n",
      "129     68   68      Active site   \n",
      "130    166  166      Active site   \n",
      "131    232  234     Binding site   \n",
      "132     75  121   Disulfide bond   \n",
      "133     19   19  Natural variant   \n",
      "134     37   37  Natural variant   \n",
      "135     67   67  Natural variant   \n",
      "136    102  102  Natural variant   \n",
      "137    162  162  Natural variant   \n",
      "138    162  162  Natural variant   \n",
      "139    235  235  Natural variant   \n",
      "140    236  236  Natural variant   \n",
      "141    237  237  Natural variant   \n",
      "142    261  261  Natural variant   \n",
      "143    272  272  Natural variant   \n",
      "144     25   38            Helix   \n",
      "145     40   48      Beta strand   \n",
      "146     49   51             Turn   \n",
      "147     54   59      Beta strand   \n",
      "148     67   69            Helix   \n",
      "149     70   83            Helix   \n",
      "150     97   99            Helix   \n",
      "151    107  109            Helix   \n",
      "152    111  113             Turn   \n",
      "153    117  126            Helix   \n",
      "154    130  139            Helix   \n",
      "155    143  152            Helix   \n",
      "156    166  168            Helix   \n",
      "157    171  173             Turn   \n",
      "158    181  193            Helix   \n",
      "159    194  197      Beta strand   \n",
      "160    199  210            Helix   \n",
      "161    213  215      Beta strand   \n",
      "162    216  218             Turn   \n",
      "163    219  222            Helix   \n",
      "164    228  235      Beta strand   \n",
      "165    237  239             Turn   \n",
      "166    241  249      Beta strand   \n",
      "167    255  263      Beta strand   \n",
      "168    264  266      Beta strand   \n",
      "169    268  284            Helix   \n",
      "170     37  201            Chain   \n",
      "171     37  112           Repeat   \n",
      "172    116  201           Repeat   \n",
      "173     66   78   Disulfide bond   \n",
      "174    145  167   Disulfide bond   \n",
      "175     42   47            Helix   \n",
      "176     54   61            Helix   \n",
      "177     63   65            Helix   \n",
      "178     66   69      Beta strand   \n",
      "179     70   72            Helix   \n",
      "180     75   83      Beta strand   \n",
      "181     86   98      Beta strand   \n",
      "182    103  111      Beta strand   \n",
      "183    121  126            Helix   \n",
      "184    133  140            Helix   \n",
      "185    143  151      Beta strand   \n",
      "186    153  156             Turn   \n",
      "187    162  168      Beta strand   \n",
      "188    170  172             Turn   \n",
      "189    175  177      Beta strand   \n",
      "190    180  188      Beta strand   \n",
      "191    191  199      Beta strand   \n",
      "\n",
      "                                              feature  \n",
      "0                                  Beta-lactamase TEM  \n",
      "1                             Acyl-ester intermediate  \n",
      "2                                     Proton acceptor  \n",
      "3                                                      \n",
      "4                                                      \n",
      "5                                            in TEM-4  \n",
      "6           in TEM-2, TEM-3, TEM-8, TEM-16 and TEM-24  \n",
      "7                                            in IRT-4  \n",
      "8    in TEM-3, TEM-4, TEM-6, TEM-8, TEM-16 and TEM-24  \n",
      "9                                 in TEM-6 and TEM-16  \n",
      "10                         in TEM-5, TEM-8 and TEM-24  \n",
      "11                                in TEM-5 and TEM-24  \n",
      "12                          in TEM-3, TEM-4 and TEM-8  \n",
      "13                                in TEM-5 and TEM-24  \n",
      "14                                           in TEM-4  \n",
      "15                                           in IRT-4  \n",
      "16                                                     \n",
      "17                                                     \n",
      "18                                                     \n",
      "19                                                     \n",
      "20                                                     \n",
      "21                                                     \n",
      "22                                                     \n",
      "23                                                     \n",
      "24                                                     \n",
      "25                                                     \n",
      "26                                                     \n",
      "27                                                     \n",
      "28                                                     \n",
      "29                                                     \n",
      "30                                                     \n",
      "31                                                     \n",
      "32                                                     \n",
      "33                                                     \n",
      "34                                                     \n",
      "35                                                     \n",
      "36                                                     \n",
      "37                                                     \n",
      "38                                                     \n",
      "39                                                     \n",
      "40                                                     \n",
      "41                                                     \n",
      "42                  Beta-lactamase inhibitory protein  \n",
      "43                                                  1  \n",
      "44                                                  2  \n",
      "45                                                     \n",
      "46                                                     \n",
      "47                                                     \n",
      "48                                                     \n",
      "49                                                     \n",
      "50                                                     \n",
      "51                                                     \n",
      "52                                                     \n",
      "53                                                     \n",
      "54                                                     \n",
      "55                                                     \n",
      "56                                                     \n",
      "57                                                     \n",
      "58                                                     \n",
      "59                                                     \n",
      "60                                                     \n",
      "61                                                     \n",
      "62                                                     \n",
      "63                                                     \n",
      "64                                 Beta-lactamase TEM  \n",
      "65                            Acyl-ester intermediate  \n",
      "66                                    Proton acceptor  \n",
      "67                                                     \n",
      "68                                                     \n",
      "69                                           in TEM-4  \n",
      "70          in TEM-2, TEM-3, TEM-8, TEM-16 and TEM-24  \n",
      "71                                           in IRT-4  \n",
      "72   in TEM-3, TEM-4, TEM-6, TEM-8, TEM-16 and TEM-24  \n",
      "73                                in TEM-6 and TEM-16  \n",
      "74                         in TEM-5, TEM-8 and TEM-24  \n",
      "75                                in TEM-5 and TEM-24  \n",
      "76                          in TEM-3, TEM-4 and TEM-8  \n",
      "77                                in TEM-5 and TEM-24  \n",
      "78                                           in TEM-4  \n",
      "79                                           in IRT-4  \n",
      "80                                                     \n",
      "81                                                     \n",
      "82                                                     \n",
      "83                                                     \n",
      "84                                                     \n",
      "85                                                     \n",
      "86                                                     \n",
      "87                                                     \n",
      "88                                                     \n",
      "89                                                     \n",
      "90                                                     \n",
      "91                                                     \n",
      "92                                                     \n",
      "93                                                     \n",
      "94                                                     \n",
      "95                                                     \n",
      "96                                                     \n",
      "97                                                     \n",
      "98                                                     \n",
      "99                                                     \n",
      "100                                                    \n",
      "101                                                    \n",
      "102                                                    \n",
      "103                                                    \n",
      "104                                                    \n",
      "105                                                    \n",
      "106                 Beta-lactamase inhibitory protein  \n",
      "107                                                 1  \n",
      "108                                                 2  \n",
      "109                                                    \n",
      "110                                                    \n",
      "111                                                    \n",
      "112                                                    \n",
      "113                                                    \n",
      "114                                                    \n",
      "115                                                    \n",
      "116                                                    \n",
      "117                                                    \n",
      "118                                                    \n",
      "119                                                    \n",
      "120                                                    \n",
      "121                                                    \n",
      "122                                                    \n",
      "123                                                    \n",
      "124                                                    \n",
      "125                                                    \n",
      "126                                                    \n",
      "127                                                    \n",
      "128                                Beta-lactamase TEM  \n",
      "129                           Acyl-ester intermediate  \n",
      "130                                   Proton acceptor  \n",
      "131                                                    \n",
      "132                                                    \n",
      "133                                          in TEM-4  \n",
      "134         in TEM-2, TEM-3, TEM-8, TEM-16 and TEM-24  \n",
      "135                                          in IRT-4  \n",
      "136  in TEM-3, TEM-4, TEM-6, TEM-8, TEM-16 and TEM-24  \n",
      "137                               in TEM-6 and TEM-16  \n",
      "138                        in TEM-5, TEM-8 and TEM-24  \n",
      "139                               in TEM-5 and TEM-24  \n",
      "140                         in TEM-3, TEM-4 and TEM-8  \n",
      "141                               in TEM-5 and TEM-24  \n",
      "142                                          in TEM-4  \n",
      "143                                          in IRT-4  \n",
      "144                                                    \n",
      "145                                                    \n",
      "146                                                    \n",
      "147                                                    \n",
      "148                                                    \n",
      "149                                                    \n",
      "150                                                    \n",
      "151                                                    \n",
      "152                                                    \n",
      "153                                                    \n",
      "154                                                    \n",
      "155                                                    \n",
      "156                                                    \n",
      "157                                                    \n",
      "158                                                    \n",
      "159                                                    \n",
      "160                                                    \n",
      "161                                                    \n",
      "162                                                    \n",
      "163                                                    \n",
      "164                                                    \n",
      "165                                                    \n",
      "166                                                    \n",
      "167                                                    \n",
      "168                                                    \n",
      "169                                                    \n",
      "170                 Beta-lactamase inhibitory protein  \n",
      "171                                                 1  \n",
      "172                                                 2  \n",
      "173                                                    \n",
      "174                                                    \n",
      "175                                                    \n",
      "176                                                    \n",
      "177                                                    \n",
      "178                                                    \n",
      "179                                                    \n",
      "180                                                    \n",
      "181                                                    \n",
      "182                                                    \n",
      "183                                                    \n",
      "184                                                    \n",
      "185                                                    \n",
      "186                                                    \n",
      "187                                                    \n",
      "188                                                    \n",
      "189                                                    \n",
      "190                                                    \n",
      "191                                                    \n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "# uniprot annotation\n",
    "annotationES = pd.read_csv('annotationES.txt', sep=' ',header=None)\n",
    "annotationES.columns = ['start' ,'end']\n",
    "#print(annotationES)\n",
    "annotationT = pd.read_csv('annotationT.txt', sep=',',header=None)\n",
    "annotationT.columns = ['type']\n",
    "#print(annotationT)\n",
    "# Open the file with read-only mode\n",
    "with open('annotationF.txt', 'r') as file:\n",
    "    lines = file.readlines()\n",
    "# Create an empty list to store the data\n",
    "annotationF = []\n",
    "# Iterate over each line\n",
    "for line in lines:\n",
    "    # Split the line on the comma to get the values\n",
    "    values = line.split(' ,')\n",
    "    # Strip leading and trailing whitespace characters from each string\n",
    "    values = [val.strip() for val in values]\n",
    "    # Append the values to the data list\n",
    "    annotationF.append(values)\n",
    "# Create a DataFrame from the data list\n",
    "annotationFdf = pd.DataFrame(annotationF,columns=['feature'])\n",
    "# Print the DataFrame\n",
    "\n",
    "\n",
    "annotationdf = pd.concat([annotationES, annotationT, annotationFdf],axis=1)\n",
    "\n",
    "print(annotationdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "f4cb0a21",
   "metadata": {},
   "outputs": [],
   "source": [
    "import Bio\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "\n",
    "#more NCBI optional data, not important for the pipeline\n",
    "def protein_to_gene(protein_id):\n",
    "    Entrez.email = \"maximschwarzberg@gmail.com\"  # Always tell NCBI who you are\n",
    "    handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "    record = Bio.SeqIO.read(handle, \"genbank\")\n",
    "    #print(record)\n",
    "    handle.close()\n",
    "    return record.annotations[\"gene_id\"]\n",
    "\n",
    "#gene_id = protein_to_gene('P62593')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4993ee79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# query protein sequence input\n",
    "from Bio import SeqIO\n",
    "protein_seq = SeqIO.read(\"sequence.842.fasta\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7fdc9c64",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('HPETLVKVKDAEDQLGARVGYIELDLNSGKILESFRPEERFPMMSTFKVLLCGA...KHW')"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_seq.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fae34407",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run an online blast search in the NCBI database\n",
    "from Bio.Blast import NCBIWWW\n",
    "result_handle = NCBIWWW.qblast(\"tblastn\", \"pdb\", protein_seq.seq) # pdb - protein data bank"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "96f8e777",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the ncbi search results as an xml document\n",
    "with open(\"TEM-1nt.xml\", \"w\") as save_to:\n",
    "    save_to.write(result_handle.read())\n",
    "    result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9e1b1844",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<_io.StringIO object at 0x7f2ad9908ee0>\n"
     ]
    }
   ],
   "source": [
    "from Bio.Blast import NCBIWWW\n",
    "from Bio import SeqIO\n",
    "\n",
    "# read in the FASTA file\n",
    "record = SeqIO.read(\"sequence.842.fasta\", \"fasta\")\n",
    "\n",
    "# run tBLASTn using the sequence and return the results as XML\n",
    "result_handle = NCBIWWW.qblast(\"tblastn\", \"nr\", record.format(\"fasta\"))\n",
    "print(result_handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f4df03d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"TEM-1nt.xml\", \"w\") as save_to:\n",
    "    save_to.write(result_handle.read())\n",
    "    result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "75d6b82a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   dna_sequence_id\n",
      "0         MN158355\n",
      "1         MN045293\n",
      "2         MN096661\n",
      "3         MF593881\n",
      "4         MF593878\n",
      "5         MF370950\n",
      "6         MF370949\n",
      "7         MF370948\n",
      "8         MF370946\n",
      "9         MF095057\n",
      "10        KY702578\n",
      "11        OL744252\n",
      "12        OL744243\n",
      "13        OP142513\n",
      "14        OK037529\n",
      "15        OK037528\n",
      "16        OK037527\n",
      "17        OK037526\n",
      "18        OK037525\n",
      "19        OK037524\n",
      "20        OK037523\n",
      "21        OK037522\n",
      "22        OK037521\n",
      "23        OK037520\n",
      "24        OK037519\n",
      "25        OK037518\n",
      "26        OK037517\n",
      "27        OK037516\n",
      "28        OK037515\n",
      "29        OK037514\n",
      "30        OK037513\n",
      "31        OK037512\n",
      "32        OK037511\n",
      "33        OK037510\n",
      "34        OK037509\n",
      "35        OK037508\n",
      "36        OK037507\n",
      "37        OK037506\n",
      "38        KJ544246\n",
      "39        KJ544243\n",
      "40        KF906263\n",
      "41        MZ026155\n",
      "42        MZ026154\n",
      "43        MZ026153\n",
      "44        MZ026151\n",
      "45        MZ026150\n",
      "46        MZ026149\n",
      "47        MZ026148\n",
      "48        MZ026146\n",
      "49        MZ026145\n"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"dna_sequence_id\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1nt.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "\n",
    "    if child.tag == 'Hit_accession':\n",
    "        Hit_accession=child.text.strip()\n",
    "\n",
    "# append data of interest into the database        \n",
    "        rows.append({\"dna_sequence_id\": Hit_accession})\n",
    "# create a dataframe and visualize it\n",
    "dna_sequence_iddf = pd.DataFrame(rows, columns=cols)\n",
    "print(dna_sequence_iddf) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "17cd3f29",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# convert the xml to a dataframe\n",
    "\n",
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "from Bio import Entrez\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"accession-nucleotide\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1nt.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "    if child.tag == 'Hit_accession':\n",
    "        Hit_accession=child.text.strip()\n",
    "        Entrez.email = \"maximschwarzberg@gmail.com\" # Always tell NCBI who you are\n",
    "        handle = Entrez.efetch(db=\"nucleotide\", id=Hit_accession, rettype=\"fasta\")\n",
    "        dna_sequence = handle.read()\n",
    "        #print(dna_sequence)\n",
    "        with open(\"accessionnt.txt\", \"a\") as f:\n",
    "            print(dna_sequence, file=f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "99cf9d3c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              nucleotide_sequence\n",
      "0                           TTACCAATGCTTAATCAGTGAGGCACCTATCTCAGCGATCTGTCTATTTCGTTCATCCATAGTTGCCTGACTCCCCGTCGTGTAGATAACTACGATACGGGAGGGCTTACCATCTGGCCCCAGTGCTGCAATGATACCGCGAGACCCACGCTCACCGGCTCCAGATTTATCAGCAATAAACCAGCCAGCCGGAAGGGCCGAGCGCAGAAGTGGTCCTGCAACTTTATCCGCCTCCATCCAGTCTATTAATTGTTGCCGGGAAGCTAGAGTAAGTAGTTCGCCAGTTAATAGTTTGCGCAACGTTGTTGCCATTGCTGCAGGCATCGTGGTGTCACGCTCGTCGTTTGGTATGGCTTCATTCAGCTCCGGTTCCCAACGATCAAGGCGAGTTACATGATCCCCCATGTTGTGCAAAAAAGCGGTTAGCTCCTTCGGTCCTCCGATCGTTGTCAGAAGTAAGTTGGCAGCAGTGTTATCACTCATGGTTATGGCAGCACTGCATAATTCTCTTACTGTCATGCCATCCGTAAGATGCTTTTCTGTGACTGGTGAGTACTCAACCAAGTCATTCTGAGAATAGTGTATGCGGCGACCGAGTTGCTCTTGCCCGGCGTCAACACGGGATAATACCGCACCACATAGCAGAACTTTAAAAGTGCTCATCATTGGAAAACGTTCTTCGGGGCGAAAACTCTCAAGGATCTTACCGCTGTTGAGATCCAGTTCGATGTAACCCACTCGTGCACCCAACTGATCTTCAGCATCTTTTACTTTCACCAGCGTTTCGGGGTGAGCAAAAACAGGAAGGCAAAATGCCGCAAAAAAGGGAATAAGGGC\n",
      "1     ATGAGTATTCAACATTTCCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGT\n",
      "2    ATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAACTGTC\n",
      "3                               TTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGCGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCGGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGAAACTGTCACAC\n",
      "4                                          GCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGAAACTG\n",
      "5      ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGATGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "6      ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "7      ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTACGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "8      ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCTGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "9      ATGAGTATTCAACATTTCCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGCGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCGGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "10   ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTA\n",
      "11  ATGAGTATTCAACATTTCCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "12  ATGAGTATTCAACATTTCCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "13  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "14  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "15  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "16  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "17  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "18  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "19  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "20  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "21  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "22  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "23  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "24  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "25  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "26  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "27  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "28  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "29  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "30  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "31  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "32  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "33  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "34  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "35  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "36  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "37  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "38                                                        TGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "39                 CATTTCCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGG\n",
      "40                          TTACCAATGCTTAATCAGTGAGGCACCTATCTCAGCGATCTGTCTATTTCGTTCATCCATAGTTGCCTGACTCCCCGTCGTGTAGATAACTACGATACGGGAGGGCTTACCATCTGGCCCCAGTGCTGCAATGATACCGCGAGACCCACGCTCACCGGCTCCAGATTTATCAGCAATAAACCAGCCAGCCGGAAGGGCCGAGCGCAGAAGTGGTCCTGCAACTTTATCCGCCTCCATCCAGTCTATTAATTGTTGCCGGGAAGCTAGAGTAAGTAGTTCGCCAGTTAATAGTTTGCGCAACGTTGTTGCCATTGCTGCAGGCATCGTGGTGTCACGCTCGTCGTTTGGTATGGCTTCATTCAGCTCCGGTTCCCAACGATCAAGGCGAGTTACATGATCCCCCATGTTGTGCAAAAAAGCGGTTAGCTCCTTCGGTCCTCCGATCGTTGTCAGAAGTAAGTTGGCAGCAGTGTTATCACTCATGGTTATGGCAGCACTGCATAATTCTCTTACTGTCATGCCATCCGTAAGATGCTTTTCTGTGACTGGTGAGTACTCAACCAAGTCATTCTGAGAATAGTGTATGCGGCGACCGAGTTGCTCTTGCCCGGCGTCAACACGGGATAATACCGCACCACATAGCAGAACTTTAAAAGTGCTCATCATTGGAAAACGTTCTTCGGGGCGAAAACTCTCAAGGATCTTACCGCTGTTGAGATCCAGTTCGATGTAACCCACTCGTGCACCCAACTGATCTTCAGCATCTTTTACTTTCACCAGCGTTTCGGGGTGAGCAAAAACAGGAAGGCAAAATGCCGCAAAAAAGGGAATAAGGGC\n",
      "41  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "42  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "43  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "44  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "45  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "46  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "47  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "48  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n",
      "49  ATGAGTATTCAACATTTTCGTGTCGCCCTTATTCCCTTTTTTGCGGCATTTTGCCTTCCTGTTTTTGCTCACCCAGAAACGCTGGTGAAAGTAAAAGATGCTGAAGATCAGTTGGGTGCACGAGTGGGTTACATCGAACTGGATCTCAACAGCGGTAAGATCCTTGAGAGTTTTCGCCCCGAAGAACGTTTTCCAATGATGAGCACTTTTAAAGTTCTGCTATGTGGTGCGGTATTATCCCGTGTTGACGCCGGGCAAGAGCAACTCGGTCGCCGCATACACTATTCTCAGAATGACTTGGTTGAGTACTCACCAGTCACAGAAAAGCATCTTACGGATGGCATGACAGTAAGAGAATTATGCAGTGCTGCCATAACCATGAGTGATAACACTGCTGCCAACTTACTTCTGACAACGATCGGAGGACCGAAGGAGCTAACCGCTTTTTTGCACAACATGGGGGATCATGTAACTCGCCTTGATCGTTGGGAACCGGAGCTGAATGAAGCCATACCAAACGACGAGCGTGACACCACGATGCCTGCAGCAATGGCAACAACGTTGCGCAAACTATTAACTGGCGAACTACTTACTCTAGCTTCCCGGCAACAATTAATAGACTGGATGGAGGCGGATAAAGTTGCAGGACCACTTCTGCGCTCGGCCCTTCCGGCTGGCTGGTTTATTGCTGATAAATCTGGAGCCGGTGAGCGTGGGTCTCGCGGTATCATTGCAGCACTGGGGCCAGATGGTAAGCCCTCCCGTATCGTAGTTATCTACACGACGGGGAGTCAGGCAACTATGGATGAACGAAATAGACAGATCGCTGAGATAGGTGCCTCACTGATTAAGCATTGGTAA\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import pandas as pd\n",
    "\n",
    "sequences = []\n",
    "# open the FASTA file\n",
    "with open('accessionnt.txt', 'r') as file:\n",
    "    # iterate over the records in the file\n",
    "    for record in SeqIO.parse(file, 'fasta'):\n",
    "        sequences.append({'nucleotide_sequence':record.seq})\n",
    "\n",
    "sequencedf = pd.DataFrame(sequences)\n",
    "sequencedf['nucleotide_sequence'] = sequencedf['nucleotide_sequence'].apply(lambda x: ''.join(str(i) for i in x))\n",
    "\n",
    "print(sequencedf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "08491f5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import sdRDM package\n",
    "try:\n",
    "    from sdRDM import DataModel\n",
    "    from sdRDM.database import build_sql_database\n",
    "except ModuleNotFoundError:\n",
    "    # Install package if not given\n",
    "    import subprocess\n",
    "    import sys\n",
    "    subprocess.call([sys.executable, \"-m\", \"pip\", \"install\", \"git+https://github.com/JR-1991/software-driven-rdm.git\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "0663df65",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ProteinSequence\n",
      "├── id\n",
      "├── protein_sequence_id\n",
      "├── name\n",
      "├── amino_acid_sequence\n",
      "├── protein_database_id\n",
      "│   └── ProteinDatabase\n",
      "│       ├── id\n",
      "│       ├── database\n",
      "│       └── link_to_database\n",
      "└── protein_organism_id\n",
      "    └── Organism\n",
      "        ├── id\n",
      "        ├── organism_name\n",
      "        └── ncbi_taxonomy_id\n",
      "DNASequence\n",
      "├── id\n",
      "├── protein_sequence_id\n",
      "│   └── ProteinSequence\n",
      "│       ├── id\n",
      "│       ├── protein_sequence_id\n",
      "│       ├── name\n",
      "│       ├── amino_acid_sequence\n",
      "│       ├── protein_database_id\n",
      "│       │   └── ProteinDatabase\n",
      "│       │       ├── id\n",
      "│       │       ├── database\n",
      "│       │       └── link_to_database\n",
      "│       └── protein_organism_id\n",
      "│           └── Organism\n",
      "│               ├── id\n",
      "│               ├── organism_name\n",
      "│               └── ncbi_taxonomy_id\n",
      "├── dna_sequence_id\n",
      "├── nucleotide_sequence\n",
      "├── dna_database_id\n",
      "│   └── DNADatabase\n",
      "│       ├── id\n",
      "│       ├── database\n",
      "│       └── link_to_database\n",
      "└── dna_organism_id\n",
      "    └── Organism\n",
      "        ├── id\n",
      "        ├── organism_name\n",
      "        └── ncbi_taxonomy_id\n",
      "StandardNumberingScheme\n",
      "├── id\n",
      "├── protein_sequence_id\n",
      "│   └── ProteinSequence\n",
      "│       ├── id\n",
      "│       ├── protein_sequence_id\n",
      "│       ├── name\n",
      "│       ├── amino_acid_sequence\n",
      "│       ├── protein_database_id\n",
      "│       │   └── ProteinDatabase\n",
      "│       │       ├── id\n",
      "│       │       ├── database\n",
      "│       │       └── link_to_database\n",
      "│       └── protein_organism_id\n",
      "│           └── Organism\n",
      "│               ├── id\n",
      "│               ├── organism_name\n",
      "│               └── ncbi_taxonomy_id\n",
      "├── standard_numering_scheme_id\n",
      "└── standard_numering_scheme_name\n"
     ]
    }
   ],
   "source": [
    "#import the data base\n",
    "pyeed = DataModel.from_git(\"https://github.com/maxim945/PyEED-Fullprototype.git\")\n",
    "#visualize for each separate table\n",
    "pyeed.ProteinSequence.visualize_tree()\n",
    "pyeed.DNASequence.visualize_tree()\n",
    "pyeed.StandardNumberingScheme.visualize_tree()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "5ecba5bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#build the data base\n",
    "build_sql_database(pyeed.ProteinSequence, loc=\"./TEM-1.db\")\n",
    "build_sql_database(pyeed.DNASequence, loc=\"./TEM-1.db\")\n",
    "build_sql_database(pyeed.StandardNumberingScheme, loc=\"./TEM-1.db\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e2c9d6bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the xml to a dataframe\n",
    "\n",
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"protein_sequence_id\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "\n",
    "    if child.tag == 'Hit_accession':\n",
    "        Hit_accession=child.text.strip()\n",
    "\n",
    "# append data of interest into the database        \n",
    "        rows.append({\"protein_sequence_id\": Hit_accession})\n",
    "# create a dataframe and visualize it\n",
    "protein_sequence_iddf = pd.DataFrame(rows, columns=cols)\n",
    "#df12.reset_index\n",
    "\n",
    "#print(protein_sequence_iddf) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "5faf973d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the xml to a dataframe\n",
    "\n",
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"name\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "\n",
    "    if child.tag == 'Hit_def':\n",
    "        Hit_def=child.text.strip()\n",
    "\n",
    "# append data of interest into the database        \n",
    "        rows.append({\"name\": Hit_def})\n",
    "# create a dataframe and visualize it\n",
    "namedf = pd.DataFrame(rows, columns=cols)\n",
    "#print(namedf) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "dfc6d87e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the xml to a dataframe\n",
    "\n",
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"amino_acid_sequence\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "\n",
    "    if child.tag == 'Hsp_midline':\n",
    "        Hsp_midline=child.text.strip()\n",
    "\n",
    "# append data of interest into the database        \n",
    "        rows.append({\"amino_acid_sequence\": Hsp_midline})\n",
    "# create a dataframe and visualize it\n",
    "amino_acid_sequencedf = pd.DataFrame(rows, columns=cols)\n",
    "#print(amino_acid_sequencedf) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "45461024",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a dataframe with 50 rows and 1 column\n",
    "protein_database_iddf = pd.DataFrame([\"PDB\"]*50, columns=[\"protein_database_id_id\"])\n",
    "\n",
    "# Print the first 5 rows of the dataframe\n",
    "#print(protein_database_iddf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e3f643e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "#open the txt file from the above cell\n",
    "taxonomy_id = pd.read_csv('taxo.txt', sep=' ',header=None)\n",
    "#create a dataframe\n",
    "taxonomy_id.columns = [\"protein_organism_id_id\", ]\n",
    "#extract the 10 smallest resolutions\n",
    "taxonomy_id = taxonomy_id['protein_organism_id_id']\n",
    "#extract the 10 smallest PDB ids which correlate to the resolution above \n",
    "\n",
    "#with resolution\n",
    "#print(taxonomy_id)\n",
    "#only 10 ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f79cc633",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "\n",
    "# Connecting to the database\n",
    "connection = sqlite3.connect('TEM-1.db')\n",
    "merged_df = pd.concat([protein_sequence_iddf, namedf,amino_acid_sequencedf,protein_database_iddf,taxonomy_id], axis=1)\n",
    "#print(merged_df)\n",
    "# Insert the dataframe into the database\n",
    "merged_df.to_sql('ProteinSequence', connection, if_exists='append', index=False)\n",
    "\n",
    "# Committing the changes\n",
    "connection.commit()\n",
    " \n",
    "# closing the database connection\n",
    "connection.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "c587562a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0        562\n",
      "1        562\n",
      "2        562\n",
      "3        562\n",
      "4        562\n",
      "5        562\n",
      "6        562\n",
      "7        562\n",
      "8        562\n",
      "9        562\n",
      "10       562\n",
      "11       562\n",
      "12       562\n",
      "13       562\n",
      "14       562\n",
      "15       562\n",
      "16       562\n",
      "17       562\n",
      "18       562\n",
      "19       562\n",
      "20       562\n",
      "21       562\n",
      "22       562\n",
      "23       562\n",
      "24       562\n",
      "25       562\n",
      "26       562\n",
      "27       562\n",
      "28       562\n",
      "29       573\n",
      "30       562\n",
      "31       562\n",
      "32       562\n",
      "33       562\n",
      "34       562\n",
      "35    299586\n",
      "36       582\n",
      "37       562\n",
      "38       562\n",
      "39       562\n",
      "40       562\n",
      "41       562\n",
      "42       562\n",
      "43       562\n",
      "44       562\n",
      "45       562\n",
      "46    469008\n",
      "47       562\n",
      "48       562\n",
      "49       562\n",
      "Name: dna_organism_id_id, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "#open the txt file from the above cell\n",
    "taxonomy_id = pd.read_csv('taxo.txt', sep=' ',header=None)\n",
    "#create a dataframe\n",
    "taxonomy_id.columns = [\"dna_organism_id_id\", ]\n",
    "#extract the 10 smallest resolutions\n",
    "taxonomy_id = taxonomy_id['dna_organism_id_id']\n",
    "#extract the 10 smallest PDB ids which correlate to the resolution above \n",
    "\n",
    "#with resolution\n",
    "print(taxonomy_id)\n",
    "#only 10 ids\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "c219db1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a dataframe with 50 rows and 1 column\n",
    "dna_database_iddf = pd.DataFrame([\"NCBI\"]*50, columns=[\"dna_database_id_id\"])\n",
    "\n",
    "# Print the first 5 rows of the dataframe\n",
    "#print(dna_database_iddf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "fc6fd0b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the xml to a dataframe\n",
    "\n",
    "import xml.etree.ElementTree as ET\n",
    "import csv\n",
    "import xml.etree.ElementTree as Xet\n",
    "import pandas as pd\n",
    "\n",
    "# chose your headers, which columns you are interested in, in your database.\n",
    "cols = [\"protein_sequence_id_id\"]\n",
    "rows = []\n",
    "\n",
    "\n",
    "# PARSE XML\n",
    "tree = ET.parse('TEM-1.xml')\n",
    "root = tree.getroot()\n",
    "\n",
    "# search for the specific hits in the xml file\n",
    "for child in root.iter():\n",
    "\n",
    "    if child.tag == 'Hit_accession':\n",
    "        Hit_accession=child.text.strip()\n",
    "\n",
    "# append data of interest into the database        \n",
    "        rows.append({\"protein_sequence_id_id\": Hit_accession})\n",
    "# create a dataframe and visualize it\n",
    "protein_sequence_iddf = pd.DataFrame(rows, columns=cols)\n",
    "#df12.reset_index\n",
    "\n",
    "#print(protein_sequence_iddf) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "bbaeebc5",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "\n",
    "# Connecting to the database\n",
    "connection = sqlite3.connect('TEM-1.db')\n",
    "merged_df = pd.concat([protein_sequence_iddf, dna_sequence_iddf,sequencedf,dna_database_iddf,taxonomy_id], axis=1)\n",
    "#print(merged_df)\n",
    "# Insert the dataframe into the database\n",
    "merged_df.to_sql('DNASequence', connection, if_exists='append', index=False)\n",
    "\n",
    "# Committing the changes\n",
    "connection.commit()\n",
    " \n",
    "# closing the database connection\n",
    "connection.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "d0530210",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Create a dataframe with 50 rows and 1 column\n",
    "standard_numering_scheme_iddf = pd.DataFrame([\"1\"], columns=[\"standard_numering_scheme_id\"])\n",
    "standard_numering_scheme_namedf = pd.DataFrame([\"4ZJ1_A\"], columns=[\"standard_numering_scheme_name\"])\n",
    "\n",
    "# Print the first 5 rows of the dataframe\n",
    "#print(dna_database_iddf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21a295fa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f1bf436",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "\n",
    "# Connecting to the database\n",
    "connection = sqlite3.connect('TEM-1.db')\n",
    "merged_df = pd.concat([protein_sequence_iddf, standard_numering_scheme_iddf,standard_numering_scheme_namedf,combined_df], axis=1)\n",
    "#print(merged_df)\n",
    "# Insert the dataframe into the database\n",
    "merged_df.to_sql('StandardNumberingScheme', connection, if_exists='append', index=False)\n",
    "\n",
    "# Committing the changes\n",
    "connection.commit()\n",
    " \n",
    "# closing the database connection\n",
    "connection.close()"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
